---
title: "Presentation"
author: "Emily Möller (s251702), Erine Genilloud (s252025), Mazarine Chambon (s251918), Yann Ducrest (s251889), Hedi Khemakhem (s251900)"
format:
  revealjs:
    theme: moon
    css: styles.css
editor: visual
class: center, middle
---

## Load

```{r}
library("tidyverse")
```

```{r}
phenotype <- readRDS("../data/phenotype_clean.rds")
gen_exp <- readRDS("../data/gen_exp_clean.rds")
#data_augmented <- readRDS("../data/data_augmented.rds")
```

### Article

**Comparison of gene expression patterns across twelve tumor types identifies a cancer supercluster characterized by TP53 mutations and cell cycle defects**

### Dataset

#### [Gene expression dataset]{.underline}

-   TCGA ovarian cancer cohort

-   **FPKM** = *Fragments Per Kilobase of transcript per Million mapped reads* → normalized gene-expression metric

-   **Format:** TSV (`TCGA-OV.star_fpkm.tsv`)

-   **Dimensions:** **60 660 genes** × **430 samples**

#### [Clinical dataset]{.underline}

-   Patient phenotype / clinical variables from TCGA-OV

-   **Format:** TSV (`TCGA-OV.clinical.tsv`)

-   **Dimensions:** **632 patients** × **84 clinical variables**

### Source

**The Cancer Genome Atlas (TCGA) – Ovarian cancer collection**

## Clean 1 : Gene table

Too many genes in the dataset for analysis (60 600 genes)

1.  Delete all genes with no expression for each sample\
2.  Find non-relevant genes:
    -   Calculate the sum of expression for each gene\
    -   Plot the sum to choose a threshold\
    -   Remove genes under this threshold

→ **Final number of genes: 14 157**

```{r}
knitr::kable(head(gen_exp, 8))
```

## Clean

phenotype table :

-   remove some variables (repetitive or useless)

```{r}
knitr::kable(head(phenotype, 8))
```

## Augment

joining of the datasets

## Describe

![](images/age.png)

## Analysis

```{r}
#| echo: true      # show code
#| eval: false     # don’t run it
5

```

```{r}
#| echo: true      # show code
#| eval: true     # run it
5

```
