<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Full Analysis Pipeline</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00_all_files/libs/clipboard/clipboard.min.js"></script>
<script src="00_all_files/libs/quarto-html/quarto.js"></script>
<script src="00_all_files/libs/quarto-html/popper.min.js"></script>
<script src="00_all_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00_all_files/libs/quarto-html/anchor.min.js"></script>
<link href="00_all_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00_all_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00_all_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00_all_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00_all_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Full Analysis Pipeline</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Download files from websource</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readr"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Download files only if missing</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">"../data/_raw/TCGA-OV.star_fpkm.tsv.gz"</span>)) {</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">"../data/_raw"</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  urls <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-OV.clinical.tsv.gz"</span>,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-OV.star_fpkm.tsv.gz"</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"../data/_raw"</span>, <span class="fu">basename</span>(urls))</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(urls)) {</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Downloading "</span>, <span class="fu">basename</span>(urls[i]), <span class="st">" ..."</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">download.file</span>(urls[i], files[i])</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/_raw/TCGA-OV.star_fpkm.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 60660 Columns: 430
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr   (1): Ensembl_ID
dbl (429): TCGA-24-1104-01A, TCGA-25-1320-01A, TCGA-24-1850-01A, TCGA-61-210...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/_raw/TCGA-OV.clinical.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 632 Columns: 84
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (65): sample, id, disease_type, case_id, submitter_id, primary_site, alc...
dbl (17): age_at_index.demographic, days_to_birth.demographic, year_of_birth...
lgl  (2): oct_embedded.samples, is_ffpe.samples

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save lighter versions for later steps</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(phenotype, <span class="st">"../data/phenotype.rds"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gen_exp, <span class="st">"../data/gen_exp.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="load-libraries" class="level1">
<h1>Load libraries</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="clean-gene-table" class="level1">
<h1>Clean gene table</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp.rds"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 60660   430</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="sc">|&gt;</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 430
   Ensembl_ID         `TCGA-24-1104-01A` `TCGA-25-1320-01A` `TCGA-24-1850-01A`
   &lt;chr&gt;                           &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000258935.1              0                   0                  0    
 2 ENSG00000280197.1              0                   0                  0    
 3 ENSG00000143196.5              0.157               1.50               0.577
 4 ENSG00000286657.1              0.0551              0.237              0    
 5 ENSG00000179611.3              0.698               0.478              0.330
 6 ENSG00000236004.2              0                   0                  0.114
 7 ENSG00000170231.16             0.0236              1.75               4.78 
 8 ENSG00000075975.17             2.42                2.05               3.46 
 9 ENSG00000178537.10             3.74                2.93               3.04 
10 ENSG00000252877.1              0                   0                  0    
# ℹ 426 more variables: `TCGA-61-2101-01A` &lt;dbl&gt;, `TCGA-23-1111-01A` &lt;dbl&gt;,
#   `TCGA-25-2396-01A` &lt;dbl&gt;, `TCGA-30-1714-01A` &lt;dbl&gt;,
#   `TCGA-04-1347-01A` &lt;dbl&gt;, `TCGA-59-2351-01A` &lt;dbl&gt;,
#   `TCGA-09-1659-01B` &lt;dbl&gt;, `TCGA-24-1431-01A` &lt;dbl&gt;,
#   `TCGA-31-1951-01A` &lt;dbl&gt;, `TCGA-29-1769-01A` &lt;dbl&gt;,
#   `TCGA-13-0916-01A` &lt;dbl&gt;, `TCGA-61-1721-01A` &lt;dbl&gt;,
#   `TCGA-61-2008-02A` &lt;dbl&gt;, `TCGA-20-1687-01A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>We have 60600 observations (rows) and 430 variables (columns). The rows are the different genes and the columns are the samples. We have 60600 genes (that is too much) and we can see that some of them are 0 for all the samples. So we want to delete all the gene with no expression for each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="sc">-</span>Ensembl_ID)) <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 57825   430</code></pre>
</div>
</div>
<p>We steel have too much genes (57825) so we want to see which genes are unuseful</p>
<p>We calculate the sum of all the gene expression for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_gene_exp =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="sc">-</span>Ensembl_ID))) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sum_gene_exp, <span class="at">.after =</span> Ensembl_ID) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sum_gene_exp)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="sc">|&gt;</span> </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 431
   Ensembl_ID         sum_gene_exp `TCGA-24-1104-01A` `TCGA-25-1320-01A`
   &lt;chr&gt;                     &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000228181.1         32.5              0                  0.0828
 2 ENSG00000225791.7        467.               0.487              0.667 
 3 ENSG00000145685.14      1028.               1.95               2.80  
 4 ENSG00000213424.9          6.19             0                  0     
 5 ENSG00000252756.1         58.8              0.451              0.197 
 6 ENSG00000140623.14        27.8              0.0126             0.0596
 7 ENSG00000082641.16      2010.               4.87               4.62  
 8 ENSG00000287406.1         58.4              0                  0.142 
 9 ENSG00000232320.6        597.               1.37               2.92  
10 ENSG00000231134.1         28.3              0                  0.0877
# ℹ 427 more variables: `TCGA-24-1850-01A` &lt;dbl&gt;, `TCGA-61-2101-01A` &lt;dbl&gt;,
#   `TCGA-23-1111-01A` &lt;dbl&gt;, `TCGA-25-2396-01A` &lt;dbl&gt;,
#   `TCGA-30-1714-01A` &lt;dbl&gt;, `TCGA-04-1347-01A` &lt;dbl&gt;,
#   `TCGA-59-2351-01A` &lt;dbl&gt;, `TCGA-09-1659-01B` &lt;dbl&gt;,
#   `TCGA-24-1431-01A` &lt;dbl&gt;, `TCGA-31-1951-01A` &lt;dbl&gt;,
#   `TCGA-29-1769-01A` &lt;dbl&gt;, `TCGA-13-0916-01A` &lt;dbl&gt;,
#   `TCGA-61-1721-01A` &lt;dbl&gt;, `TCGA-61-2008-02A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Now we want to visualise this new varaible to know what treshold we are going to use to select the relevant genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-15-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-15-2.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">y =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-15-3.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can see that a lot of genes have the sum around 0 (with the boxplot). In the histogram, the binwidth is 100 and 4 bars are higher than the others and correspond to irelevant genes. So we can choose a treshold of 400. There are 431 samples so a threshold of 400 means that in average, apromimativelly all the genes expressions are higher than 1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_gene_exp <span class="sc">&gt;</span> <span class="dv">400</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 14157   431</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-17-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Now, we have 14157 genes which is more appropriate for our case.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(sum_gene_exp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gen_exp_clean, <span class="st">"../data/gen_exp_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-phenotype-table" class="level1">
<h1>Cleaning Phenotype table</h1>
<section id="loading-table" class="level2">
<h2 class="anchored" data-anchor-id="loading-table">Loading table</h2>
<p>We start by loading the “phenotype” table, with the raw clinical data. The slice_sample allows to see 20 random samples and start evaluating which variables (columns) will be useful</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype.rds"</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 84
   sample           id            disease_type case_id submitter_id primary_site
   &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;        &lt;chr&gt;   &lt;chr&gt;        &lt;chr&gt;       
 1 TCGA-25-1322-01A b520aae5-65a… Cystic, Muc… b520aa… TCGA-25-1322 Ovary       
 2 TCGA-30-1714-01A 09820a33-80b… Cystic, Muc… 09820a… TCGA-30-1714 Ovary       
 3 TCGA-13-1405-01A 29dccd25-4c4… Cystic, Muc… 29dccd… TCGA-13-1405 Ovary       
 4 TCGA-25-1315-01A 635f5335-b00… Cystic, Muc… 635f53… TCGA-25-1315 Ovary       
 5 TCGA-61-2611-02A 499645bd-e33… Cystic, Muc… 499645… TCGA-61-2611 Ovary       
 6 TCGA-24-0975-01A 6a57948c-0dc… Cystic, Muc… 6a5794… TCGA-24-0975 Ovary       
 7 TCGA-42-2593-01A 3db7ecdd-315… Cystic, Muc… 3db7ec… TCGA-42-2593 Ovary       
 8 TCGA-29-1784-01A 23d32627-9fd… Cystic, Muc… 23d326… TCGA-29-1784 Ovary       
 9 TCGA-61-2612-01A c040df8a-9d6… Cystic, Muc… c040df… TCGA-61-2612 Ovary       
10 TCGA-61-2094-01A b31d85cc-d2e… Cystic, Muc… b31d85… TCGA-61-2094 Ovary       
11 TCGA-25-1328-01A 79fd602b-3e8… Cystic, Muc… 79fd60… TCGA-25-1328 Ovary       
12 TCGA-61-2610-02A 1098441d-fb1… Cystic, Muc… 109844… TCGA-61-2610 Ovary       
13 TCGA-13-0723-01A ff530f28-0ec… Cystic, Muc… ff530f… TCGA-13-0723 Ovary       
14 TCGA-29-1690-01A 6b06281b-d3a… Cystic, Muc… 6b0628… TCGA-29-1690 Ovary       
15 TCGA-04-1367-01A 30b8f4cd-924… Cystic, Muc… 30b8f4… TCGA-04-1367 Ovary       
16 TCGA-13-A5FU-01A 0f7ac6ca-780… Cystic, Muc… 0f7ac6… TCGA-13-A5FU Ovary       
17 TCGA-24-2020-01A e20ea417-729… Cystic, Muc… e20ea4… TCGA-24-2020 Ovary       
18 TCGA-24-2036-01A c9226204-cb6… Cystic, Muc… c92262… TCGA-24-2036 Ovary       
19 TCGA-25-1878-01A 9bdbaccf-b43… Cystic, Muc… 9bdbac… TCGA-25-1878 Ovary       
20 TCGA-29-2429-01A 4309ca23-1cf… Cystic, Muc… 4309ca… TCGA-29-2429 Ovary       
# ℹ 78 more variables: alcohol_history.exposures &lt;chr&gt;, race.demographic &lt;chr&gt;,
#   gender.demographic &lt;chr&gt;, ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   primary_site.project &lt;chr&gt;, project_id.project &lt;chr&gt;,
#   disease_type.project &lt;chr&gt;, name.project &lt;chr&gt;, …</code></pre>
</div>
</div>
<section id="removal-of-variables" class="level3">
<h3 class="anchored" data-anchor-id="removal-of-variables">Removal of variables</h3>
<p>We start by removing some unuseful variables, such as:</p>
<ul>
<li><p>“id”, “case_id”, “tissue_source_site_id.tissue_source_site”, “entity_id.annotations”, “annotation_id.annotations” and other:<br>
They are long lists of seemingly random characters, useful for identification of the sample but not for analysis, as we have no information on their meaning.<br>
We’ll use “sample” as identification.</p></li>
<li><p>“submitter_id”, “entity_submitter_id.annotations” :<br>
They are the same as “sample” without the last 3 characters. This is repetitive and untidy.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows how some variables repeat the same information </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(sample, submitter_id)) <span class="sc">|&gt;</span> </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   sample           submitter_id
   &lt;chr&gt;            &lt;chr&gt;       
 1 TCGA-09-0366-01A TCGA-09-0366
 2 TCGA-61-1722-01A TCGA-61-1722
 3 TCGA-72-4240-01A TCGA-72-4240
 4 TCGA-61-1895-01A TCGA-61-1895
 5 TCGA-04-1525-01A TCGA-04-1525
 6 TCGA-29-1769-01A TCGA-29-1769
 7 TCGA-13-A5FU-01A TCGA-13-A5FU
 8 TCGA-61-2087-01A TCGA-61-2087
 9 TCGA-04-1353-11B TCGA-04-1353
10 TCGA-24-1417-01A TCGA-24-1417
11 TCGA-23-1021-01B TCGA-23-1021
12 TCGA-24-0970-01B TCGA-24-0970
13 TCGA-61-1730-01A TCGA-61-1730
14 TCGA-57-1992-01A TCGA-57-1992
15 TCGA-30-1862-01A TCGA-30-1862
16 TCGA-25-2391-01A TCGA-25-2391
17 TCGA-23-1114-01B TCGA-23-1114
18 TCGA-23-1023-01R TCGA-23-1023
19 TCGA-24-1852-01A TCGA-24-1852
20 TCGA-5X-AA5U-01A TCGA-5X-AA5U</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows identifier variables, no meaning for analysis </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(id, case_id)) <span class="sc">|&gt;</span> </span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   id                                   case_id                             
   &lt;chr&gt;                                &lt;chr&gt;                               
 1 3b81ec7c-0934-4649-9231-9919dd26dd15 3b81ec7c-0934-4649-9231-9919dd26dd15
 2 f58e49fc-9641-4b46-be4c-766445ec70a3 f58e49fc-9641-4b46-be4c-766445ec70a3
 3 5201ee13-2aed-4641-9169-4d5ee07a23da 5201ee13-2aed-4641-9169-4d5ee07a23da
 4 e9b5336e-d724-4e7d-8c81-1147abd0a80d e9b5336e-d724-4e7d-8c81-1147abd0a80d
 5 f3618472-32dc-4c90-a407-90050f68be85 f3618472-32dc-4c90-a407-90050f68be85
 6 ee0a4a13-613e-4c5d-96c3-8083a013702d ee0a4a13-613e-4c5d-96c3-8083a013702d
 7 914d8613-3c25-4a10-be50-28b42f4d3a5d 914d8613-3c25-4a10-be50-28b42f4d3a5d
 8 fe0e3851-d8cb-4533-9536-b4826cd25f87 fe0e3851-d8cb-4533-9536-b4826cd25f87
 9 60014a6e-6a13-4f92-bbfd-eeeee9632f98 60014a6e-6a13-4f92-bbfd-eeeee9632f98
10 c183d3fb-2eee-44f8-890e-b9bf907141e6 c183d3fb-2eee-44f8-890e-b9bf907141e6
11 c0c3caab-9277-4a31-a96c-c607e38d5ccc c0c3caab-9277-4a31-a96c-c607e38d5ccc
12 15fc17cd-d073-4f81-a4a2-2db3cb9d7069 15fc17cd-d073-4f81-a4a2-2db3cb9d7069
13 1d9893c8-0de3-4b07-b0ba-a53019b23eb4 1d9893c8-0de3-4b07-b0ba-a53019b23eb4
14 02594e5e-8751-47c1-9245-90c66984b665 02594e5e-8751-47c1-9245-90c66984b665
15 1db60f09-7f5a-4f21-8003-06a6abc781db 1db60f09-7f5a-4f21-8003-06a6abc781db
16 9bf16a89-2fc7-4c08-93bc-3105eec5c3cc 9bf16a89-2fc7-4c08-93bc-3105eec5c3cc
17 30c7efdd-5a86-4418-a33c-a5d27fe56e42 30c7efdd-5a86-4418-a33c-a5d27fe56e42
18 dfb7dc5d-92f4-4a3a-9667-db38efcc9fdb dfb7dc5d-92f4-4a3a-9667-db38efcc9fdb
19 0d23fa5b-95f8-4626-a4d2-c72ad3cc553b 0d23fa5b-95f8-4626-a4d2-c72ad3cc553b
20 b783f33b-734a-4801-bf39-b559a2035c6f b783f33b-734a-4801-bf39-b559a2035c6f</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these variables </span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype <span class="sc">|&gt;</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(id, </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>            case_id, </span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>            submitter_id, </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>            tissue_source_site_id.tissue_source_site, </span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>            entity_submitter_id.annotations, </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>            entity_id.annotations, </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>            annotation_id.annotations, </span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>            case_id.annotations, </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>            treatment_id.treatments.diagnoses, </span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>            submitter_id.treatments.diagnoses, </span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>            sample_id.samples, </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>            pathology_report_uuid.samples)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>         )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>“primary_site”, “primary_site.project”, “project_id.project”, … : These columns contain the same observation for each row, and will not be useful for analysis either.<br>
For example “gender.demographic” will always be “female”, as we are studying ovarian cancer.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows variables with repeated observations </span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(primary_site, </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>           name.program.project, </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>           last_known_disease_status.diagnoses, </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>           gender.demographic)) <span class="sc">|&gt;</span> </span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 4
   primary_site name.program.project last_known_disease_sta…¹ gender.demographic
   &lt;chr&gt;        &lt;chr&gt;                &lt;chr&gt;                    &lt;chr&gt;             
 1 Ovary        TCGA                 not reported             female            
 2 Ovary        TCGA                 not reported             female            
 3 Ovary        TCGA                 not reported             female            
 4 Ovary        TCGA                 not reported             female            
 5 Ovary        TCGA                 not reported             female            
 6 Ovary        TCGA                 not reported             female            
 7 Ovary        TCGA                 not reported             female            
 8 Ovary        TCGA                 not reported             female            
 9 Ovary        TCGA                 not reported             female            
10 Ovary        TCGA                 not reported             female            
11 Ovary        TCGA                 not reported             female            
12 Ovary        TCGA                 not reported             female            
13 Ovary        TCGA                 not reported             female            
14 Ovary        TCGA                 not reported             female            
15 Ovary        TCGA                 not reported             female            
16 Ovary        TCGA                 not reported             female            
17 Ovary        TCGA                 not reported             female            
18 Ovary        TCGA                 not reported             female            
19 Ovary        TCGA                 not reported             female            
20 Ovary        TCGA                 not reported             female            
# ℹ abbreviated name: ¹​last_known_disease_status.diagnoses</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these variables </span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(primary_site, </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>            primary_site.project, </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>            project_id.project, </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>            disease_type.project, </span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>            name.project, </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>            name.program.project, </span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>            project.tissue_source_site, </span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>            days_to_diagnosis.diagnoses, </span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>            last_known_disease_status.diagnoses, </span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>            tissue_or_organ_of_origin.diagnoses, </span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>            preservation_method.samples, </span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>            is_ffpe.samples, </span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>            gender.demographic)</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>         )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>“entity_type.annotations” , “state.annotations”, … : These are empty or always the same when not empty. Samples might come from different sources, which would explain irregular conventions. After careful reading of the table, we can say that the fact that some observations are left empty has no other meaning.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows variables with repeated or empty observations </span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(entity_type.annotations, </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>           state.annotations, </span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>           status.annotations)) <span class="sc">|&gt;</span> </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 3
   entity_type.annotations state.annotations status.annotations
   &lt;chr&gt;                   &lt;chr&gt;             &lt;chr&gt;             
 1 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 2 case                    released          Approved          
 3 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 4 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 5 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 6 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 7 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 8 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 9 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
10 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
11 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
12 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
13 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
14 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
15 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
16 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
17 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
18 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
19 case                    released          Approved          
20 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these columns </span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(entity_type.annotations, </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>            state.annotations, </span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>            status.annotations, </span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>            composition.samples, </span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>            days_to_sample_procurement.samples, </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>            state.treatments.diagnoses,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>            tumor_grade.diagnoses,</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>            progression_or_recurrence.diagnoses))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The samples with a disease_type value other than “Cystic, Mucinous and Serous Neoplasms” miss a lot of other information (NA in other columns). We will not keep them as there are only 8 samples and will disturb analysis without providing new information. It now makes no sense to keep this column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check for other disease types </span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(disease_type <span class="sc">!=</span> <span class="st">"Cystic, Mucinous and Serous Neoplasms"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 51
  sample           disease_type alcohol_history.exposures race.demographic
  &lt;chr&gt;            &lt;chr&gt;        &lt;chr&gt;                     &lt;chr&gt;           
1 TCGA-01-0633-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
2 TCGA-01-0628-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
3 TCGA-01-0630-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
4 TCGA-01-0631-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
5 TCGA-01-0637-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
6 TCGA-01-0636-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
7 TCGA-01-0642-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
8 TCGA-01-0639-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
# ℹ 47 more variables: ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove samples with missing values </span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(disease_type <span class="sc">==</span> <span class="st">"Cystic, Mucinous and Serous Neoplasms"</span>)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove this column</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>disease_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The column prior_treatment.diagnoses is Not Reported or No for every samples except one Yes. This column will not be useful.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show values other than "Not Reported" and "No" for prior_treatment.diagnoses</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(prior_treatment.diagnoses<span class="sc">!=</span><span class="st">"Not Reported"</span> <span class="sc">&amp;</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>         prior_treatment.diagnoses <span class="sc">!=</span> <span class="st">"No"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(prior_treatment.diagnoses, <span class="at">.before =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 50
  prior_treatment.diagnoses sample       alcohol_history.expo…¹ race.demographic
  &lt;chr&gt;                     &lt;chr&gt;        &lt;chr&gt;                  &lt;chr&gt;           
1 Yes                       TCGA-61-172… Not Reported           white           
# ℹ abbreviated name: ¹​alcohol_history.exposures
# ℹ 46 more variables: ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>Some variables don’t always contain the same information on paper, but further analysis shows us that the observations are very similar. For example alcohol_history.exposures is “Not reported” or NA:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show NAs in alcohol_history.exposures</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(alcohol_history.exposures)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11 × 50
   sample          alcohol_history.expo…¹ race.demographic ethnicity.demographic
   &lt;chr&gt;           &lt;chr&gt;                  &lt;chr&gt;            &lt;chr&gt;                
 1 TCGA-36-2539-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 2 TCGA-72-4236-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 3 TCGA-72-4237-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 4 TCGA-72-4238-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 5 TCGA-72-4235-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 6 TCGA-72-4232-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 7 TCGA-72-4241-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 8 TCGA-72-4233-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 9 TCGA-72-4240-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
10 TCGA-72-4231-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
11 TCGA-72-4234-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
# ℹ abbreviated name: ¹​alcohol_history.exposures
# ℹ 46 more variables: vital_status.demographic &lt;chr&gt;,
#   age_at_index.demographic &lt;dbl&gt;, days_to_birth.demographic &lt;dbl&gt;,
#   year_of_birth.demographic &lt;dbl&gt;, days_to_death.demographic &lt;dbl&gt;,
#   year_of_death.demographic &lt;dbl&gt;, code.tissue_source_site &lt;chr&gt;,
#   name.tissue_source_site &lt;chr&gt;, bcr_id.tissue_source_site &lt;chr&gt;,
#   notes.annotations &lt;chr&gt;, submitter_id.annotations &lt;chr&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove this column</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>alcohol_history.exposures) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Modification of the variable vital_status.demographic to a binary variable “alive”</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">vital_status.demographic =</span> <span class="fu">case_when</span>(</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>    vital_status.demographic <span class="sc">==</span> <span class="st">"Alive"</span> <span class="sc">~</span> <span class="cn">TRUE</span>,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>    vital_status.demographic <span class="sc">==</span> <span class="st">"Dead"</span> <span class="sc">~</span> <span class="cn">FALSE</span>,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  ))<span class="sc">|&gt;</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">alive=</span>vital_status.demographic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(phenotype_clean, <span class="st">"../data/phenotype_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-libraries-1" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-1">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-cleaned-tables" class="level2">
<h2 class="anchored" data-anchor-id="load-cleaned-tables">Load cleaned tables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp_clean.rds"</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="reshape-gene-expression-table" class="level2">
<h2 class="anchored" data-anchor-id="reshape-gene-expression-table">Reshape gene expression table</h2>
<p>The current gene expression dataset is in a wide format:<br>
rows = genes columns = samples<br>
For further analyses, it is more convenient to work in a long format: one row per (gene × sample) measurement. The dataset is now tidy.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>gen_exp_long <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(Ensembl_ID),</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expression"</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>               )</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>gen_exp_long <span class="sc">|&gt;</span> </span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 3
   Ensembl_ID         sample           expression
   &lt;chr&gt;              &lt;chr&gt;                 &lt;dbl&gt;
 1 ENSG00000105219.10 TCGA-31-1953-01A       1.30
 2 ENSG00000134030.14 TCGA-29-1705-01A       3.44
 3 ENSG00000266373.1  TCGA-13-0901-01B       1.11
 4 ENSG00000092068.20 TCGA-04-1356-01A       1.99
 5 ENSG00000123338.13 TCGA-04-1648-01A       1.37
 6 ENSG00000123268.9  TCGA-25-1326-01A       3.97
 7 ENSG00000167468.18 TCGA-13-1489-02A       8.87
 8 ENSG00000162772.17 TCGA-36-1571-01A       2.55
 9 ENSG00000064655.19 TCGA-29-1695-01A       4.43
10 ENSG00000080845.18 TCGA-23-1029-01B       4.18</code></pre>
</div>
</div>
</section>
<section id="join-clinical-metadata" class="level2">
<h2 class="anchored" data-anchor-id="join-clinical-metadata">Join clinical metadata</h2>
<p>Now we merge the gene expression table with the phenotype table using the common identifier sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge expression data with phenotype metadata</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> gen_exp_long <span class="sc">|&gt;</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(phenotype_clean, <span class="at">by =</span> <span class="st">"sample"</span>) <span class="sc">|&gt;</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sample, <span class="at">.before =</span> Ensembl_ID)</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>full_data <span class="sc">|&gt;</span> </span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 51
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-61-1… ENSG00000…      2.03  white            not hispanic or lati… TRUE 
 2 TCGA-24-1… ENSG00000…      2.76  black or africa… not reported          FALSE
 3 TCGA-61-1… ENSG00000…      2.10  not reported     not reported          TRUE 
 4 TCGA-13-0… ENSG00000…      1.97  white            hispanic or latino    FALSE
 5 TCGA-59-2… ENSG00000…      1.98  not reported     not reported          FALSE
 6 TCGA-61-2… ENSG00000…      4.11  white            not hispanic or lati… TRUE 
 7 TCGA-10-0… ENSG00000…      5.96  white            not reported          FALSE
 8 TCGA-23-1… ENSG00000…      0.213 white            not reported          TRUE 
 9 TCGA-25-1… ENSG00000…      3.08  white            not hispanic or lati… FALSE
10 TCGA-24-2… ENSG00000…      3.31  white            not reported          FALSE
# ℹ 45 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>We now have: biological measurements (gene-exp) + patient characteristics (phenotype).</p>
</section>
<section id="create-additional-variables" class="level2">
<h2 class="anchored" data-anchor-id="create-additional-variables">Create additional variables</h2>
<p>We can also create new useful variables like:</p>
<ul>
<li><p>A log-transformed expression variable (log2_expression), commonly used in transcriptomics to reduce skewness and easier interpretation.</p></li>
<li><p>A categorical age variable (age_cat), based on common thresholds in clinical studies.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> full_data <span class="sc">|&gt;</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log2_expression =</span> <span class="fu">log2</span>(expression <span class="sc">+</span> <span class="dv">1</span>),</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">age_cat =</span> <span class="fu">case_when</span>(</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>           age_at_index.demographic <span class="sc">&lt;</span> <span class="dv">50</span> <span class="sc">~</span> <span class="st">"young"</span>,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>           age_at_index.demographic <span class="sc">&gt;=</span> <span class="dv">50</span> <span class="sc">&amp;</span> age_at_index.demographic <span class="sc">&lt;</span> <span class="dv">65</span> <span class="sc">~</span> <span class="st">"middle"</span>,</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>           age_at_index.demographic <span class="sc">&gt;=</span> <span class="dv">65</span> <span class="sc">~</span> <span class="st">"senior"</span>,</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>           <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span>)</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>         )</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>full_data <span class="sc">|&gt;</span> </span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-24-1… ENSG00000…       3.48 white            not reported          TRUE 
 2 TCGA-24-0… ENSG00000…       1.18 white            not reported          FALSE
 3 TCGA-24-1… ENSG00000…       3.12 white            not reported          TRUE 
 4 TCGA-13-1… ENSG00000…       3.70 asian            not hispanic or lati… FALSE
 5 TCGA-29-1… ENSG00000…       6.01 white            not reported          FALSE
 6 TCGA-25-1… ENSG00000…       2.48 white            not hispanic or lati… FALSE
 7 TCGA-29-1… ENSG00000…       4.92 white            not reported          FALSE
 8 TCGA-25-1… ENSG00000…       1.40 white            not hispanic or lati… FALSE
 9 TCGA-24-2… ENSG00000…       1.20 white            not reported          FALSE
10 TCGA-31-1… ENSG00000…       4.04 white            not hispanic or lati… TRUE 
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>The +1 in the log transformation prevents issues when expression = 0. (goes to -inf with log)</p>
<p>NA_character_ ensures that missing age values remain coded as valid character-type missing values in the age category variable.</p>
</section>
<section id="save-augmented-dataset" class="level2">
<h2 class="anchored" data-anchor-id="save-augmented-dataset">Save augmented dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(full_data, <span class="st">"../data/data_augmented.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="loading-libraries-and-data" class="level3">
<h3 class="anchored" data-anchor-id="loading-libraries-and-data">0.Loading libraries and data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load 3 tables</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype_clean.rds"</span>)</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp_clean.rds"</span>)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="general-information" class="level2">
<h2 class="anchored" data-anchor-id="general-information">1. General information</h2>
<p>The analysis will be based on the article <em>Comparison of gene expression patterns across twelve tumor types identifies a cancer supercluster characterized by TP53 mutations and cell cycle defects</em>. The datasets used in the article are from the The Cancer Genome Atlas (TCGA) ovarian cancer collection. In the article, compares gene-expression profiles across multiple tumor types to identify transcriptional superclusters.</p>
<p>Two different data sets are used:</p>
<ul>
<li><p>A clinical dataset: TCGA-OV.clinical.tsv</p></li>
<li><p>A gene expression dataset: TCGA-OV.star_fpkm.tsv</p></li>
</ul>
<p>The clinical and transcriptomic datasets for the TCGA ovarian cancer cohort will be used to search for gene expression associations with clinical variables and outcomes.</p>
</section>
<section id="dataset-descriptions" class="level2">
<h2 class="anchored" data-anchor-id="dataset-descriptions">2. Dataset descriptions</h2>
<section id="clinical-dataset" class="level3">
<h3 class="anchored" data-anchor-id="clinical-dataset">Clinical dataset</h3>
<ul>
<li><p>Format: Tab-separated values (TSV) file.</p></li>
<li><p>Contains clinical variables from the TCGA ovarian cancer cohort. As it contains observations about the patients, it is called <code>phenotype</code> in this project.</p></li>
<li><p>Size: There are 632 patients as rows and 84 clinical variables as columns.</p></li>
<li><p>Key variables and data types: The following variables are most relevant for the analysis, - <code>alive</code> (bool),</p>
<ul>
<li><code>figo_stage.diagnoses</code> (chr),</li>
<li><code>morphology.diagnoses</code> (chr, the first part encodes the tumor type, the second part encodes the malignancy),</li>
<li><code>age_at_earliest_diagnosis_in_years.diagnoses.xena_derived</code> (numeric),</li>
<li><code>treatment_type.treatments.diagnoses</code> (chr, indicating the type of treatement),</li>
</ul></li>
<li><p>Needed cleaning: The dataset contains a lot of redundant variables (e.g.&nbsp;many instances of a variable indicating the ID), many have been combined into a single one or have been removed altogether. Also the variable containing information about the gender has been removed as all patients with ovarian cancer are female. Some variables contain annotations to the patients, but as most are empty and irrelevant for the analysis they have been removed as well.</p></li>
<li><p>Size after cleaning: 624 patients on the rows, 52 clinical variables on the columns.</p></li>
</ul>
<p>Underneath, 10 randomly selected rows of the set are shown to understand the structure:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> <span class="fu">slice_sample</span>(<span class="at">n=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 49
   sample    race.demographic ethnicity.demographic alive age_at_index.demogra…¹
   &lt;chr&gt;     &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;                  &lt;dbl&gt;
 1 TCGA-30-… white            not hispanic or lati… FALSE                     56
 2 TCGA-25-… white            not hispanic or lati… FALSE                     66
 3 TCGA-36-… white            not reported          TRUE                      63
 4 TCGA-25-… white            not hispanic or lati… FALSE                     68
 5 TCGA-13-… white            not hispanic or lati… FALSE                     48
 6 TCGA-29-… white            not reported          TRUE                      61
 7 TCGA-24-… white            not reported          TRUE                      45
 8 TCGA-24-… white            not reported          FALSE                     59
 9 TCGA-13-… white            not hispanic or lati… FALSE                     60
10 TCGA-29-… white            not reported          TRUE                      59
# ℹ abbreviated name: ¹​age_at_index.demographic
# ℹ 44 more variables: days_to_birth.demographic &lt;dbl&gt;,
#   year_of_birth.demographic &lt;dbl&gt;, days_to_death.demographic &lt;dbl&gt;,
#   year_of_death.demographic &lt;dbl&gt;, code.tissue_source_site &lt;chr&gt;,
#   name.tissue_source_site &lt;chr&gt;, bcr_id.tissue_source_site &lt;chr&gt;,
#   notes.annotations &lt;chr&gt;, submitter_id.annotations &lt;chr&gt;,
#   classification.annotations &lt;chr&gt;, created_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
<section id="gene-expression-dataset" class="level3">
<h3 class="anchored" data-anchor-id="gene-expression-dataset">Gene expression dataset</h3>
<ul>
<li><p>Format: Tab-separated values (TSV) file.</p></li>
<li><p>Contains gene expression data (FPKM) from the TCGA ovarian cancer cohort.</p></li>
<li><p>The values are FPKM (Fragments Per Kilobase of transcript per Million mapped reads) which is a normalized gene-expression metric.</p></li>
<li><p>Size: There are 60660 genes on the rows and 430 samples on the columns.</p></li>
<li><p>Data types: numeric non‐negative values</p></li>
<li><p>Needed cleaning: Some genes have a null expression level for all samples, these genes have been removed from the dataset. To further reduce the number of genes and keep only the most relevant ones, the summed expression level over all samples for each gene has been computed and the relevant genes were kept based on a threshold. A variable representing this sum of the expression levels for each gene has been added.</p></li>
<li><p>Size after cleaning: 14157 genes on the rows, 431 variables on the columns.</p></li>
</ul>
<p>Underneath, 10 randomly selected rows of the set are shown to understand the structure:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="sc">|&gt;</span> <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 430
   Ensembl_ID         `TCGA-24-1104-01A` `TCGA-25-1320-01A` `TCGA-24-1850-01A`
   &lt;chr&gt;                           &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000147050.16               1.10              0.999               1.05
 2 ENSG00000102359.8                1.10              2.01                1.47
 3 ENSG00000164105.4                2.60              3.53                2.88
 4 ENSG00000196972.9                2.09              1.46                1.08
 5 ENSG00000100387.9                3.97              3.68                3.70
 6 ENSG00000261373.1                2.14              2.09                1.52
 7 ENSG00000167555.14               1.46              2.23                1.28
 8 ENSG00000118873.16               2.28              3.02                2.29
 9 ENSG00000213866.3                3.61              3.43                3.89
10 ENSG00000164088.18               3.06              2.19                2.76
# ℹ 426 more variables: `TCGA-61-2101-01A` &lt;dbl&gt;, `TCGA-23-1111-01A` &lt;dbl&gt;,
#   `TCGA-25-2396-01A` &lt;dbl&gt;, `TCGA-30-1714-01A` &lt;dbl&gt;,
#   `TCGA-04-1347-01A` &lt;dbl&gt;, `TCGA-59-2351-01A` &lt;dbl&gt;,
#   `TCGA-09-1659-01B` &lt;dbl&gt;, `TCGA-24-1431-01A` &lt;dbl&gt;,
#   `TCGA-31-1951-01A` &lt;dbl&gt;, `TCGA-29-1769-01A` &lt;dbl&gt;,
#   `TCGA-13-0916-01A` &lt;dbl&gt;, `TCGA-61-1721-01A` &lt;dbl&gt;,
#   `TCGA-61-2008-02A` &lt;dbl&gt;, `TCGA-20-1687-01A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="joining-of-the-datasets" class="level3">
<h3 class="anchored" data-anchor-id="joining-of-the-datasets">Joining of the datasets</h3>
<p>These two datasets have been combined to one by matching the patient IDs (samples). As the two datasets do not show the same amount of samples (the phenotype dataset is bigger than the gene expression dataset), the IDs that could not be resolved between the two sets have been removed. This leaves only patients who have recorded gene expression data (no gene expression data is lost).</p>
<p>Two variables have been added: - <code>log2_expression</code> (numeric, gene expression on logarithmic scale) - <code>age_cat</code> (chr, age category the patient belongs to: young, middle, senior)</p>
<p>In addition to the key variables mentioned for the phenotype (clinical) data and the gene expression values from the gene expression data, these two variables will allow for easier visualization and analysis.</p>
<p>In the <code>figo_stage.diagnoses</code> variable, the tumor stages IA, IB and the majority of IC have disappeared after joining. This is due to the fact that it it rare to have gene expression data for early stage cancers.</p>
<p>Underneath, the 10 first rows of the joined set are shown to understand the structure:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> <span class="fu">head</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-24-1… ENSG00000…      1.40  white            not reported          FALSE
 2 TCGA-25-1… ENSG00000…      0.624 white            not hispanic or lati… FALSE
 3 TCGA-24-1… ENSG00000…      0.523 white            not reported          TRUE 
 4 TCGA-61-2… ENSG00000…      0.371 white            not hispanic or lati… FALSE
 5 TCGA-23-1… ENSG00000…      1.37  white            not reported          TRUE 
 6 TCGA-25-2… ENSG00000…      0.462 white            not hispanic or lati… FALSE
 7 TCGA-30-1… ENSG00000…      1.51  white            not hispanic or lati… FALSE
 8 TCGA-04-1… ENSG00000…      0.901 white            not hispanic or lati… TRUE 
 9 TCGA-59-2… ENSG00000…      0.490 white            not hispanic or lati… TRUE 
10 TCGA-09-1… ENSG00000…      0.796 white            not hispanic or lati… FALSE
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
</section>
<section id="quality-and-intergrity" class="level2">
<h2 class="anchored" data-anchor-id="quality-and-intergrity">3. Quality and intergrity</h2>
<p>In the phenotype datasets, a few columns containing only NAs have been removed. Some of the variables contained a label “Not Reported” instead of NA. As described above, many variables described redundant information and needed to be removed.</p>
<p>The gene expression dataset is of good quality and integrity, the only correction needed was to remove uninformative genes.</p>
<p>Overall the data seems of good quality without aberrant values or errors in the writing of the sets.</p>
</section>
<section id="visualisation" class="level2">
<h2 class="anchored" data-anchor-id="visualisation">4. Visualisation</h2>
<p>The following graphs will allow to visualize the content of the data set.</p>
<p>A list of the variables contained in the full dataset:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> <span class="fu">glimpse</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 6,073,353
Columns: 53
$ sample                                                    &lt;chr&gt; "TCGA-24-110…
$ Ensembl_ID                                                &lt;chr&gt; "ENSG0000027…
$ expression                                                &lt;dbl&gt; 1.3958306, 0…
$ race.demographic                                          &lt;chr&gt; "white", "wh…
$ ethnicity.demographic                                     &lt;chr&gt; "not reporte…
$ alive                                                     &lt;lgl&gt; FALSE, FALSE…
$ age_at_index.demographic                                  &lt;dbl&gt; 56, 65, 72, …
$ days_to_birth.demographic                                 &lt;dbl&gt; -20527, -240…
$ year_of_birth.demographic                                 &lt;dbl&gt; 1945, 1936, …
$ days_to_death.demographic                                 &lt;dbl&gt; 1933, 1155, …
$ year_of_death.demographic                                 &lt;dbl&gt; 2006, 2004, …
$ code.tissue_source_site                                   &lt;chr&gt; "24", "25", …
$ name.tissue_source_site                                   &lt;chr&gt; "Washington …
$ bcr_id.tissue_source_site                                 &lt;chr&gt; "IGC", "IGC"…
$ notes.annotations                                         &lt;chr&gt; NA, NA, NA, …
$ submitter_id.annotations                                  &lt;chr&gt; NA, NA, NA, …
$ classification.annotations                                &lt;chr&gt; NA, NA, NA, …
$ created_datetime.annotations                              &lt;chr&gt; NA, NA, NA, …
$ updated_datetime.annotations                              &lt;chr&gt; NA, NA, NA, …
$ category.annotations                                      &lt;chr&gt; NA, NA, NA, …
$ case_submitter_id.annotations                             &lt;chr&gt; NA, NA, NA, …
$ figo_stage.diagnoses                                      &lt;chr&gt; "Stage IV", …
$ synchronous_malignancy.diagnoses                          &lt;chr&gt; "Not Reporte…
$ days_to_last_follow_up.diagnoses                          &lt;dbl&gt; 1933, 1124, …
$ age_at_diagnosis.diagnoses                                &lt;dbl&gt; 20527, 24018…
$ primary_diagnosis.diagnoses                               &lt;chr&gt; "Serous cyst…
$ prior_malignancy.diagnoses                                &lt;chr&gt; "not reporte…
$ year_of_diagnosis.diagnoses                               &lt;dbl&gt; 2001, 2001, …
$ prior_treatment.diagnoses                                 &lt;chr&gt; "No", "No", …
$ morphology.diagnoses                                      &lt;chr&gt; "8441/3", "8…
$ classification_of_tumor.diagnoses                         &lt;chr&gt; "not reporte…
$ icd_10_code.diagnoses                                     &lt;chr&gt; "C56.9", "C5…
$ site_of_resection_or_biopsy.diagnoses                     &lt;chr&gt; "Ovary", "Ov…
$ age_at_earliest_diagnosis.diagnoses.xena_derived          &lt;dbl&gt; 20527, 24018…
$ age_at_earliest_diagnosis_in_years.diagnoses.xena_derived &lt;dbl&gt; 56.23836, 65…
$ updated_datetime.treatments.diagnoses                     &lt;chr&gt; "['2019-07-3…
$ treatment_type.treatments.diagnoses                       &lt;chr&gt; "['Radiation…
$ treatment_or_therapy.treatments.diagnoses                 &lt;chr&gt; "['yes', 'ye…
$ created_datetime.treatments.diagnoses                     &lt;chr&gt; "['', '2019-…
$ sample_type_id.samples                                    &lt;chr&gt; "01", "01", …
$ tumor_descriptor.samples                                  &lt;chr&gt; "Primary", "…
$ sample_type.samples                                       &lt;chr&gt; "Primary Tum…
$ days_to_collection.samples                                &lt;dbl&gt; NA, NA, NA, …
$ initial_weight.samples                                    &lt;dbl&gt; NA, NA, NA, …
$ intermediate_dimension.samples                            &lt;dbl&gt; 1.0, 3.0, 1.…
$ shortest_dimension.samples                                &lt;dbl&gt; 0.7, 1.0, 0.…
$ oct_embedded.samples                                      &lt;lgl&gt; NA, NA, NA, …
$ specimen_type.samples                                     &lt;chr&gt; "Unknown", "…
$ longest_dimension.samples                                 &lt;dbl&gt; 1.5, 3.4, 2.…
$ tissue_type.samples                                       &lt;chr&gt; "Tumor", "Tu…
$ annotations.samples                                       &lt;chr&gt; NA, NA, NA, …
$ log2_expression                                           &lt;dbl&gt; 1.2605259, 0…
$ age_cat                                                   &lt;chr&gt; "middle", "s…</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>data_augmented_numeric_summary <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> </span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">list</span>(<span class="at">mean =</span> <span class="sc">~</span><span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">median =</span> <span class="sc">~</span><span class="fu">median</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sd =</span> <span class="sc">~</span><span class="fu">sd</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min =</span> <span class="sc">~</span><span class="fu">min</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">max =</span> <span class="sc">~</span><span class="fu">max</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">missing =</span> <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.x))),</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">.names =</span> <span class="st">"{.col}_{.fn}"</span>)) </span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>data_augmented_numeric_summary</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 102
  expression_mean expression_median expression_sd expression_min expression_max
            &lt;dbl&gt;             &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;          &lt;dbl&gt;
1            2.80              2.53          1.68              0           16.1
# ℹ 97 more variables: expression_missing &lt;int&gt;,
#   age_at_index.demographic_mean &lt;dbl&gt;, age_at_index.demographic_median &lt;dbl&gt;,
#   age_at_index.demographic_sd &lt;dbl&gt;, age_at_index.demographic_min &lt;dbl&gt;,
#   age_at_index.demographic_max &lt;dbl&gt;, age_at_index.demographic_missing &lt;int&gt;,
#   days_to_birth.demographic_mean &lt;dbl&gt;,
#   days_to_birth.demographic_median &lt;dbl&gt;, days_to_birth.demographic_sd &lt;dbl&gt;,
#   days_to_birth.demographic_min &lt;dbl&gt;, days_to_birth.demographic_max &lt;dbl&gt;, …</code></pre>
</div>
</div>
<section id="distribution-of-the-age-at-diagnosis" class="level3">
<h3 class="anchored" data-anchor-id="distribution-of-the-age-at-diagnosis">Distribution of the age at diagnosis</h3>
<p>The graph is a histogram showing the distribution of patient age at diagnosis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>patient_level <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>    sample,</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">age =</span> age_at_earliest_diagnosis_in_years.diagnoses.xena_derived</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age))</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>patient_level <span class="sc">|&gt;</span> </span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> age)) <span class="sc">+</span></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">"skyblue"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">mean</span>(age)), <span class="at">color =</span> <span class="st">"red"</span>,</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Distribution of Age at Diagnosis (Patient Level)"</span>,</span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Age"</span>,</span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Number of Patients"</span></span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-66-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../doc/images/age.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<p>The age of patients at diagnosis seems to be normally distributed with a mean (red line) at around ~60 years.</p>
</section>
<section id="age-distribution-at-tumor-stage" class="level3">
<h3 class="anchored" data-anchor-id="age-distribution-at-tumor-stage">Age distribution at tumor stage</h3>
<p>Tumor stages IA, IB and the majority of IC have disappeared after joining the initial two datasets. Most patients, out of those which also have gene expression data, have ovarian cancer stage IIIC. There are very few data points for stages IC, IIA, IIB and IIIA.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sample, figo_stage.diagnoses) <span class="sc">|&gt;</span> </span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(figo_stage.diagnoses)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 9 × 2
  figo_stage.diagnoses     n
  &lt;chr&gt;                &lt;int&gt;
1 Stage IC                 1
2 Stage IIA                3
3 Stage IIB                5
4 Stage IIC               18
5 Stage IIIA               7
6 Stage IIIB              17
7 Stage IIIC             311
8 Stage IV                64
9 &lt;NA&gt;                     3</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse to one row per patient</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>patient_level <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>    sample,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">figo_stage =</span> figo_stage.diagnoses,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">age =</span> age_at_earliest_diagnosis_in_years.diagnoses.xena_derived</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age), <span class="sc">!</span><span class="fu">is.na</span>(figo_stage))</span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a>patient_level <span class="sc">|&gt;</span> </span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> figo_stage, <span class="at">y =</span> age)) <span class="sc">+</span></span>
<span id="cb77-13"><a href="#cb77-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">"skyblue"</span>) <span class="sc">+</span></span>
<span id="cb77-14"><a href="#cb77-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.4</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb77-15"><a href="#cb77-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb77-16"><a href="#cb77-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb77-17"><a href="#cb77-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Age Distribution by Tumor Stage (Patient Level)"</span>,</span>
<span id="cb77-18"><a href="#cb77-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Tumor Stage"</span>,</span>
<span id="cb77-19"><a href="#cb77-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Age at Diagnosis"</span></span>
<span id="cb77-20"><a href="#cb77-20" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-68-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The graph shows how age at diagnosis varies across different tumor stages. Stage IIB, IIC, IIIC, and IV show wide spreads in age, meaning patient ages vary a lot within those groups. As seen in the table above, there are few data points for stages IC, IIA, IIB and IIIA. There are some outliers, especially in Stage IIIB.</p>
</section>
<section id="boxplot-across-all-samples-of-a-set-of-genes" class="level3">
<h3 class="anchored" data-anchor-id="boxplot-across-all-samples-of-a-set-of-genes">Boxplot across all samples of a set of genes</h3>
<p>The graph displays boxplots representing the distribution of gene expression levels for five randomly selected genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pick 5 random genes</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>random_genes <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Ensembl_ID) <span class="sc">|&gt;</span> </span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">5</span>) <span class="sc">|&gt;</span> </span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID)</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the data for the chosen genes</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>subset_data <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> random_genes)</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>subset_data <span class="sc">|&gt;</span> </span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> log2_expression)) <span class="sc">+</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">"skyblue"</span>) <span class="sc">+</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.4</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Log2 Expression of Randomly Selected Genes"</span>,</span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Gene (Ensembl ID)"</span>,</span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Log2 Expression"</span></span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a>  ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-69-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The graph provides a glance on the gene expression of a selection of 5 genes.</p>
</section>
<section id="histogram-of-gene-expression-levels" class="level3">
<h3 class="anchored" data-anchor-id="histogram-of-gene-expression-levels">Histogram of gene expression levels</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2_expression)) <span class="sc">+</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">50</span>, <span class="at">fill =</span> <span class="st">"skyblue"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">mean</span>(log2_expression)), <span class="at">color =</span> <span class="st">"red"</span>,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gene Expression Distribution"</span>,</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Expression Level"</span>,</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Count"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-70-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The mean log2 gene expression level is at around 1.8. It seems to be normally distributed.</p>
</section>
</section>
<section id="loading-libraries-and-data-1" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-and-data-1">Loading libraries and data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load 3 tables</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype_clean.rds"</span>)</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp_clean.rds"</span>)</span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analysis-of-the-distribution-of-variables" class="level2">
<h2 class="anchored" data-anchor-id="analysis-of-the-distribution-of-variables">Analysis of the distribution of variables</h2>
<p><strong>FIGO Stage Distribution</strong></p>
<p>The bar plot shows the distribution of patients across FIGO stages. From the plot, it is clear that the majority of patients are classified as stage IIIC, while the numbers in other stages (stage IC, IIA, IIB, IIC, IIIA, IIIB, IIIC and IV) are considerably lower. This uneven distribution suggests that most patients in this dataset were diagnosed at an advanced stage of disease, which is common for ovarian cancers due to late detection. Because the dataset is heavily skewed toward Stage III, it is difficult to draw meaningful comparisons between stages or to identify trends related to clinical outcomes across the full spectrum of FIGO stages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">|&gt;</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">|&gt;</span></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(figo_stage.diagnoses)) <span class="sc">|&gt;</span></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> figo_stage.diagnoses)) <span class="sc">+</span></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"distribution of FIGO stages"</span>,</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"FIGO stage"</span>,</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"number of patients"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-86-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Alive or death status</strong></p>
<p>The bar plot shows the distribution of patients classified as “alive” or “dead” in the dataset. While the plot provides a visual overview of survival status, the interpretation is limited because we lack detailed information about the causes of death, the timing of events, or the follow-up duration. Without this contextual information, it is not possible to draw meaningful conclusions regarding survival patterns, treatment effectiveness, or disease progression. The data simply indicate whether a patient was recorded as alive or dead at a certain point, but do not capture the underlying clinical or temporal context necessary for a robust survival analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">|&gt;</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">|&gt;</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> alive)) <span class="sc">+</span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"distribution of alive patients"</span>,</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"alive"</span>,</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"number of patients"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-87-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="biological-relevance-of-most-variant-genes" class="level2">
<h2 class="anchored" data-anchor-id="biological-relevance-of-most-variant-genes">Biological relevance of most variant genes</h2>
<p>In this section we will extract the top 10 genes by variance of FPKM, found their biological signification and plotted expression values across all samples.</p>
<section id="finding-most-variant-genes" class="level3">
<h3 class="anchored" data-anchor-id="finding-most-variant-genes">Finding most variant genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute variance of expression per gene</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>gene_var <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">|&gt;</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">var_expression =</span> <span class="fu">var</span>(<span class="fu">c_across</span>(<span class="sc">-</span>Ensembl_ID), <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.before =</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(var_expression))</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract top 10 genes</span></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> gene_var <span class="sc">|&gt;</span> </span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">|&gt;</span> </span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID) <span class="co"># Makes genes_of_interest a character vector with genes names</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>genes_of_interest</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ENSG00000211592.8" "ENSG00000211896.7" "ENSG00000239951.1"
 [4] "ENSG00000211598.2" "ENSG00000253755.1" "ENSG00000211673.2"
 [7] "ENSG00000243466.1" "ENSG00000157005.4" "ENSG00000211677.2"
[10] "ENSG00000241351.3"</code></pre>
</div>
</div>
<p>The genes are listed using their Ensembl ID, which is a unique identifier assigned by the Ensembl projects. we will convert them to gene symbols to make them more understandable.</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Ensembl_ID</th>
<th>Gene symbol</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000211592.8</td>
<td>IGKC</td>
</tr>
<tr class="even">
<td>ENSG00000211896.7</td>
<td>IGHG1</td>
</tr>
<tr class="odd">
<td>ENSG00000239951.1</td>
<td>IGKV3-20</td>
</tr>
<tr class="even">
<td>ENSG00000211598.2</td>
<td>IGKV4-1</td>
</tr>
<tr class="odd">
<td>ENSG00000253755.1</td>
<td>IGHGP</td>
</tr>
<tr class="even">
<td>ENSG00000211673.2</td>
<td>IGLV3-1</td>
</tr>
<tr class="odd">
<td>ENSG00000243466.1</td>
<td>IGKV1-5</td>
</tr>
<tr class="even">
<td>ENSG00000157005.4</td>
<td>SST</td>
</tr>
<tr class="odd">
<td>ENSG00000211677.2</td>
<td>IGLC2</td>
</tr>
<tr class="even">
<td>ENSG00000241351.3</td>
<td>IGKV3-11</td>
</tr>
</tbody>
</table>
<p>These are all immunoglobulin genes, except for SST gene. They are expressed by B cells and plasma cells, not by tumor cells. Their high variability likely reflects differences in immune infiltration across ovarian tumor samples.</p>
<p>This suggests that immune involvement varies substantially among patients, which is consistent with known heterogeneity in the tumor microenvironment. In other words, different patients’ tumors contain different levels of immune cells, a result that is expected in ovarian cancer.</p>
</section>
<section id="expression-across-samples" class="level3">
<h3 class="anchored" data-anchor-id="expression-across-samples">Expression across samples</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot of expression</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="sc">|&gt;</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> genes_of_interest) <span class="sc">|&gt;</span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Ensembl_ID =</span> <span class="fu">factor</span>(Ensembl_ID, <span class="at">levels =</span> genes_of_interest)) <span class="sc">|&gt;</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Ensembl_ID,</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">|&gt;</span></span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> expression)) <span class="sc">+</span></span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">45</span>,        </span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="dv">1</span>,         </span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">6</span>           </span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Expression of the 10 most variable genes across patients"</span>,</span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Expression"</span></span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-90-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="relationship-with-phenotype-variables" class="level3">
<h3 class="anchored" data-anchor-id="relationship-with-phenotype-variables">Relationship with phenotype variables</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Joining the genes of interest expressions with phenotype</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>joined <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> genes_of_interest) <span class="sc">|&gt;</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Ensembl_ID,</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expr"</span>) <span class="sc">|&gt;</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">|&gt;</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">immune_score =</span> <span class="fu">mean</span>(expr)) <span class="sc">|&gt;</span></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(phenotype, <span class="at">by =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After several attempts, we found no clear relationship between these genes expressions and any patients characteristics, such as age or alive status. These plots show the lack of association between the variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune genes vs age</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(joined, <span class="fu">aes</span>(<span class="at">x =</span> immune_score, <span class="at">y =</span> age_at_index.demographic)) <span class="sc">+</span> </span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-92-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune genese vs alive/dead</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(joined, <span class="fu">aes</span>(<span class="at">x =</span> immune_score, <span class="at">y =</span> alive)) <span class="sc">+</span> </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-92-2.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune genes vs age</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(joined, <span class="fu">aes</span>(<span class="at">x =</span> immune_score, <span class="at">y =</span> race.demographic)) <span class="sc">+</span> </span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-92-3.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="aim" class="level2">
<h2 class="anchored" data-anchor-id="aim">Aim</h2>
<p>In this document, we perform an exploratory analysis of the gene expression data using Principal Component Analysis (PCA).</p>
<p>We use the augmented tidy dataset created in <code>03_augment.qmd</code> and investigate whether patients separate in PCA space according to clinical variables such as <code>alive</code> status and age category.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-augmented-data" class="level2">
<h2 class="anchored" data-anchor-id="load-augmented-data">Load augmented data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-09-1… ENSG00000…       2.56 white            not hispanic or lati… FALSE
 2 TCGA-13-0… ENSG00000…       3.75 white            not hispanic or lati… FALSE
 3 TCGA-24-1… ENSG00000…       1.54 white            not reported          TRUE 
 4 TCGA-24-2… ENSG00000…       2.27 white            not reported          FALSE
 5 TCGA-24-1… ENSG00000…       5.53 white            not reported          FALSE
 6 TCGA-10-0… ENSG00000…       3.06 white            not reported          FALSE
 7 TCGA-61-1… ENSG00000…       1.97 white            not reported          TRUE 
 8 TCGA-31-1… ENSG00000…       1.58 white            not hispanic or lati… TRUE 
 9 TCGA-61-2… ENSG00000…       3.81 white            not hispanic or lati… FALSE
10 TCGA-29-1… ENSG00000…       1.57 white            not reported          FALSE
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
<section id="select-most-variable-genes" class="level2">
<h2 class="anchored" data-anchor-id="select-most-variable-genes">Select most variable genes</h2>
<p>PCA is more informative when applied to genes that show substantial variation across samples. We therefore select the top 500 most variable genes based on the variance of <code>log2_expression</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>gene_var <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Ensembl_ID) <span class="sc">|&gt;</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">var_log2 =</span> <span class="fu">var</span>(log2_expression, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups  =</span> <span class="st">"drop"</span>) <span class="sc">|&gt;</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(var_log2))</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>gene_var <span class="sc">|&gt;</span></span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 2
   Ensembl_ID         var_log2
   &lt;chr&gt;                 &lt;dbl&gt;
 1 ENSG00000226278.1      1.42
 2 ENSG00000200197.1      1.29
 3 ENSG00000141096.6      1.23
 4 ENSG00000253755.1      1.19
 5 ENSG00000134184.13     1.19
 6 ENSG00000270372.1      1.16
 7 ENSG00000277739.1      1.16
 8 ENSG00000167244.21     1.13
 9 ENSG00000211662.2      1.10
10 ENSG00000211935.3      1.10</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> gene_var <span class="sc">|&gt;</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">500</span>) <span class="sc">|&gt;</span></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(top_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 500</code></pre>
</div>
</div>
</section>
<section id="build-expression-matrix-for-pca" class="level2">
<h2 class="anchored" data-anchor-id="build-expression-matrix-for-pca">Build expression matrix for PCA</h2>
<p>We construct a wide expression matrix with one row per sample and one column per gene (for the selected top 500 genes). We also keep a separate table with clinical covariates per sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>expr_wide <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> top_genes) <span class="sc">|&gt;</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, Ensembl_ID, log2_expression) <span class="sc">|&gt;</span></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from  =</span> Ensembl_ID,<span class="at">values_from =</span> log2_expression)</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>pheno_pca <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, alive, age_cat, age_at_index.demographic) <span class="sc">|&gt;</span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>expr_wide <span class="ot">&lt;-</span> expr_wide <span class="sc">|&gt;</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pheno_pca, <span class="at">by =</span> <span class="st">"sample"</span>)</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>expr_wide <span class="sc">|&gt;</span></span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 504
  sample           ENSG00000197632.9 ENSG00000167634.12 ENSG00000243236.6
  &lt;chr&gt;                        &lt;dbl&gt;              &lt;dbl&gt;             &lt;dbl&gt;
1 TCGA-24-1104-01A             0.597              0.478            0.0410
2 TCGA-25-1320-01A             1.03               1.13             0.0647
3 TCGA-24-1850-01A             1.31               2.45             2.89  
4 TCGA-61-2101-01A             1.12               1.82             0.551 
5 TCGA-23-1111-01A             0.497              0.223            0.210 
# ℹ 500 more variables: ENSG00000211655.3 &lt;dbl&gt;, ENSG00000043355.12 &lt;dbl&gt;,
#   ENSG00000211593.2 &lt;dbl&gt;, ENSG00000170627.11 &lt;dbl&gt;,
#   ENSG00000169550.14 &lt;dbl&gt;, ENSG00000231476.1 &lt;dbl&gt;, ENSG00000282600.2 &lt;dbl&gt;,
#   ENSG00000282122.1 &lt;dbl&gt;, ENSG00000244699.1 &lt;dbl&gt;, ENSG00000264063.1 &lt;dbl&gt;,
#   ENSG00000181449.4 &lt;dbl&gt;, ENSG00000211972.2 &lt;dbl&gt;, ENSG00000259439.2 &lt;dbl&gt;,
#   ENSG00000105997.22 &lt;dbl&gt;, ENSG00000258484.4 &lt;dbl&gt;,
#   ENSG00000096006.12 &lt;dbl&gt;, ENSG00000249395.4 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="run-pca" class="level2">
<h2 class="anchored" data-anchor-id="run-pca">Run PCA</h2>
<p>We extract the numeric gene expression columns, centre and scale them, and run PCA with <code>prcomp</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> expr_wide <span class="sc">|&gt;</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>sample, <span class="sc">-</span>alive, <span class="sc">-</span>age_cat, <span class="sc">-</span>age_at_index.demographic) <span class="sc">|&gt;</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr_mat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 429 500</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>pca_res <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x      =</span> expr_mat,</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">center =</span> <span class="cn">TRUE</span>,</span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>var_explained <span class="ot">&lt;-</span> pca_res<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pca_res<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>var_explained[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.18880878 0.06278369 0.04250761 0.03633852 0.03106406</code></pre>
</div>
</div>
</section>
<section id="scree-plot" class="level2">
<h2 class="anchored" data-anchor-id="scree-plot">Scree plot</h2>
<p>We visualise how much variance is explained by the first principal components.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>scree_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC =</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="fu">seq_along</span>(var_explained)),</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">variance =</span> var_explained)</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>scree_plot <span class="ot">&lt;-</span> scree_df <span class="sc">|&gt;</span></span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">15</span>) <span class="sc">|&gt;</span></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y =</span> variance, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Variance explained by principal components"</span>,</span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Principal component"</span>,</span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Proportion of variance explained"</span>) <span class="sc">+</span></span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-18"><a href="#cb105-18" aria-hidden="true" tabindex="-1"></a>scree_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-108-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">"../results/06_pca_scree.png"</span>,</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> scree_plot,</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This suggests that most of the global variability in the gene expression profiles is driven by a single dominant axis (PC1). Biological separation between clinical groups, if present, is therefore expected to appear mainly along PC1 or PC2.</p>
</section>
<section id="pc1-vs-pc2-coloured-by-clinical-variables" class="level2">
<h2 class="anchored" data-anchor-id="pc1-vs-pc2-coloured-by-clinical-variables">PC1 vs PC2 coloured by clinical variables</h2>
<p>We project each sample onto the first two principal components and colour points by <code>alive</code> and shape by <code>age_cat</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(pca_res<span class="sc">$</span>x, <span class="at">rownames =</span> <span class="st">"row_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(expr_wide <span class="sc">|&gt;</span></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(sample, alive, age_cat, age_at_index.demographic)) <span class="sc">|&gt;</span></span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sample)</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">|&gt;</span></span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 434
  sample      row_id   PC1    PC2   PC3   PC4   PC5    PC6    PC7     PC8    PC9
  &lt;chr&gt;       &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;
1 TCGA-24-11… 1       7.42 -8.70   1.33 -2.09  3.92 -2.51   2.86  -6.48    2.11 
2 TCGA-25-13… 2       9.39  3.18  -1.64  5.39 -3.90  0.937 -0.183  1.14    0.877
3 TCGA-24-18… 3       5.65  6.87  -3.47 -1.19  1.35  1.49  -6.81  -0.0208 -4.47 
4 TCGA-61-21… 4      -2.67  1.63  -5.82  1.74 -3.75 -2.57  -3.73   1.19    3.36 
5 TCGA-23-11… 5      -9.49 -0.285 -3.06  2.31  2.30  0.682 -0.588 -1.34    3.16 
# ℹ 423 more variables: PC10 &lt;dbl&gt;, PC11 &lt;dbl&gt;, PC12 &lt;dbl&gt;, PC13 &lt;dbl&gt;,
#   PC14 &lt;dbl&gt;, PC15 &lt;dbl&gt;, PC16 &lt;dbl&gt;, PC17 &lt;dbl&gt;, PC18 &lt;dbl&gt;, PC19 &lt;dbl&gt;,
#   PC20 &lt;dbl&gt;, PC21 &lt;dbl&gt;, PC22 &lt;dbl&gt;, PC23 &lt;dbl&gt;, PC24 &lt;dbl&gt;, PC25 &lt;dbl&gt;,
#   PC26 &lt;dbl&gt;, PC27 &lt;dbl&gt;, PC28 &lt;dbl&gt;, PC29 &lt;dbl&gt;, PC30 &lt;dbl&gt;, PC31 &lt;dbl&gt;,
#   PC32 &lt;dbl&gt;, PC33 &lt;dbl&gt;, PC34 &lt;dbl&gt;, PC35 &lt;dbl&gt;, PC36 &lt;dbl&gt;, PC37 &lt;dbl&gt;,
#   PC38 &lt;dbl&gt;, PC39 &lt;dbl&gt;, PC40 &lt;dbl&gt;, PC41 &lt;dbl&gt;, PC42 &lt;dbl&gt;, PC43 &lt;dbl&gt;,
#   PC44 &lt;dbl&gt;, PC45 &lt;dbl&gt;, PC46 &lt;dbl&gt;, PC47 &lt;dbl&gt;, PC48 &lt;dbl&gt;, PC49 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>pc1_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, scales<span class="sc">::</span><span class="fu">percent</span>(var_explained[<span class="dv">1</span>]), <span class="st">")"</span>)</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>pc2_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, scales<span class="sc">::</span><span class="fu">percent</span>(var_explained[<span class="dv">2</span>]), <span class="st">")"</span>)</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>pc1_pc2_plot <span class="ot">&lt;-</span> scores <span class="sc">|&gt;</span></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1,</span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> PC2,</span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> alive,</span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> age_cat)) <span class="sc">+</span></span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA on top 500 most variable genes"</span>,</span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Colour: alive status, shape: age category"</span>,</span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> pc1_label,</span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> pc2_label) <span class="sc">+</span></span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a>pc1_pc2_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-111-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">"../results/06_pca_pc1_pc2.png"</span>,</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> pc1_pc2_plot,</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>There is no clear separation between alive and deceased patients along PC1 or PC2. Both groups are widely mixed across the PCA space. Similarly, the age categories do not form distinct clusters, and their points are evenly spread. Overall, the first two principal components capture variability in gene expression, but this variability does not appear to be driven by the clinical variables considered here.</p>
</section>
<section id="genes-contributing-most-to-pc1-and-pc2" class="level2">
<h2 class="anchored" data-anchor-id="genes-contributing-most-to-pc1-and-pc2">Genes contributing most to PC1 and PC2</h2>
<p>Finally, we inspect which genes contribute most to PC1 and PC2 by looking at the PCA loadings.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>loadings <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>  pca_res<span class="sc">$</span>rotation,</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">rownames =</span> <span class="st">"Ensembl_ID"</span>)</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>top_pc1 <span class="ot">&lt;-</span> loadings <span class="sc">|&gt;</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Ensembl_ID, PC1) <span class="sc">|&gt;</span></span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(PC1))) <span class="sc">|&gt;</span></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>top_pc2 <span class="ot">&lt;-</span> loadings <span class="sc">|&gt;</span></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Ensembl_ID, PC2) <span class="sc">|&gt;</span></span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(PC2))) <span class="sc">|&gt;</span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>top_pc1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   Ensembl_ID            PC1
   &lt;chr&gt;               &lt;dbl&gt;
 1 ENSG00000244437.1  0.0960
 2 ENSG00000211949.3  0.0952
 3 ENSG00000239951.1  0.0952
 4 ENSG00000132465.12 0.0950
 5 ENSG00000211660.3  0.0949
 6 ENSG00000211653.2  0.0949
 7 ENSG00000224373.3  0.0949
 8 ENSG00000211938.2  0.0948
 9 ENSG00000211945.2  0.0948
10 ENSG00000211668.2  0.0948
11 ENSG00000211648.2  0.0948
12 ENSG00000211947.2  0.0946
13 ENSG00000211966.2  0.0946
14 ENSG00000211964.3  0.0946
15 ENSG00000211651.3  0.0946
16 ENSG00000241351.3  0.0945
17 ENSG00000243466.1  0.0945
18 ENSG00000211893.4  0.0944
19 ENSG00000270550.1  0.0940
20 ENSG00000211962.2  0.0938</code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>top_pc2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   Ensembl_ID            PC2
   &lt;chr&gt;               &lt;dbl&gt;
 1 ENSG00000137285.10  0.118
 2 ENSG00000092758.18  0.117
 3 ENSG00000149948.14  0.113
 4 ENSG00000142549.9   0.112
 5 ENSG00000232835.1   0.111
 6 ENSG00000148735.15 -0.111
 7 ENSG00000231295.1  -0.109
 8 ENSG00000124440.16  0.108
 9 ENSG00000171236.10 -0.108
10 ENSG00000236253.7  -0.107
11 ENSG00000142449.13  0.106
12 ENSG00000198075.10  0.105
13 ENSG00000122756.15  0.105
14 ENSG00000184697.7   0.104
15 ENSG00000189377.9  -0.103
16 ENSG00000276644.5   0.101
17 ENSG00000130294.17  0.101
18 ENSG00000112319.19  0.101
19 ENSG00000187634.12  0.100
20 ENSG00000258815.1   0.100</code></pre>
</div>
</div>
<p>The genes with the largest absolute loadings on PC1 and PC2 are those that drive the main variation in the dataset. They could be further investigated in a biological context.</p>
</section>
<section id="conclusion" class="level2">
<h2 class="anchored" data-anchor-id="conclusion">Conclusion</h2>
<p>PCA on the top 500 most variable genes reveals that the major sources of transcriptomic variance do not correspond to clinical variables such as survival status or age. The first two PCs capture global variability in gene expression, but this variability appears unrelated to the metadata included here. Genes with the highest loadings drive these main axes of variation and could be investigated further to understand the underlying biological processes.</p>
</section>
<section id="aim-1" class="level2">
<h2 class="anchored" data-anchor-id="aim-1">Aim</h2>
<p>In this document we perform a simple differential expression analysis to investigate whether gene expression differs between patients who are alive versus dead at follow-up.</p>
<p>We use the augmented tidy dataset created in <code>03_augment.qmd</code>.</p>
</section>
<section id="setup-1" class="level2">
<h2 class="anchored" data-anchor-id="setup-1">Setup</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-augmented-data-1" class="level2">
<h2 class="anchored" data-anchor-id="load-augmented-data-1">Load augmented data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>full_data <span class="sc">|&gt;</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-13-0… ENSG00000…      2.22  white            not hispanic or lati… TRUE 
 2 TCGA-29-2… ENSG00000…      0.860 white            not reported          TRUE 
 3 TCGA-13-1… ENSG00000…      3.56  white            not hispanic or lati… FALSE
 4 TCGA-13-0… ENSG00000…      2.22  white            not hispanic or lati… TRUE 
 5 TCGA-09-1… ENSG00000…      2.19  white            not hispanic or lati… TRUE 
 6 TCGA-24-1… ENSG00000…      5.14  white            not reported          TRUE 
 7 TCGA-09-2… ENSG00000…      2.15  white            not hispanic or lati… TRUE 
 8 TCGA-24-1… ENSG00000…      3.73  white            not reported          FALSE
 9 TCGA-09-0… ENSG00000…      2.14  white            not reported          FALSE
10 TCGA-24-1… ENSG00000…      0.382 white            not reported          FALSE
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
<section id="filter-to-samples-with-known-vital-status" class="level2">
<h2 class="anchored" data-anchor-id="filter-to-samples-with-known-vital-status">Filter to samples with known vital status</h2>
<p>We first restrict the dataset to samples with a non-missing <code>alive</code> status and check the number of samples in each group.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>full_data_alive <span class="ot">&lt;-</span> full_data <span class="sc">|&gt;</span> </span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(alive))</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>full_data_alive <span class="sc">|&gt;</span></span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sample, alive) <span class="sc">|&gt;</span></span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(alive)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 2
  alive     n
  &lt;lgl&gt; &lt;int&gt;
1 FALSE   265
2 TRUE    164</code></pre>
</div>
</div>
</section>
<section id="differential-expression-per-gene" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-per-gene">Differential expression per gene</h2>
<p>For each gene, we compare log2-transformed expression between patients who are alive (<code>alive == TRUE</code>) and dead (<code>alive == FALSE</code>) using a two-sample t-test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> full_data_alive <span class="sc">|&gt;</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Ensembl_ID) <span class="sc">|&gt;</span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_alive =</span> <span class="fu">mean</span>(log2_expression[alive], <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_dead =</span> <span class="fu">mean</span>(log2_expression[<span class="sc">!</span>alive], <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">log2_fc =</span> mean_alive <span class="sc">-</span> mean_dead, </span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_value =</span> <span class="fu">tryCatch</span>(<span class="fu">t.test</span>(log2_expression <span class="sc">~</span> alive)<span class="sc">$</span>p.value, <span class="at">error =</span> <span class="cf">function</span>(e) <span class="cn">NA_real_</span>), </span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">|&gt;</span></span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_adj =</span> <span class="fu">p.adjust</span>(p_value, <span class="at">method =</span> <span class="st">"BH"</span>))</span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-10"><a href="#cb120-10" aria-hidden="true" tabindex="-1"></a>de_results <span class="sc">|&gt;</span></span>
<span id="cb120-11"><a href="#cb120-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 6
   Ensembl_ID         mean_alive mean_dead  log2_fc p_value p_adj
   &lt;chr&gt;                   &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;
 1 ENSG00000000003.15       2.38      2.35  0.0311  0.0790  0.508
 2 ENSG00000000419.13       2.70      2.69  0.0126  0.251   0.695
 3 ENSG00000000457.14       1.35      1.31  0.0435  0.0208  0.370
 4 ENSG00000000460.17       1.26      1.18  0.0781  0.00287 0.275
 5 ENSG00000000938.13       1.28      1.19  0.0828  0.0339  0.404
 6 ENSG00000000971.16       1.67      1.72 -0.0486  0.293   0.721
 7 ENSG00000001036.14       2.28      2.30 -0.0177  0.295   0.723
 8 ENSG00000001084.13       1.57      1.52  0.0504  0.0296  0.395
 9 ENSG00000001167.14       2.07      2.03  0.0448  0.0264  0.387
10 ENSG00000001460.18       1.19      1.20 -0.00681 0.761   0.933</code></pre>
</div>
</div>
<p>We save the full table of differential expression results for downstream use.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(de_results, <span class="st">"../results/07_de_results_alive_vs_dead.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="summary-of-differential-expression" class="level2">
<h2 class="anchored" data-anchor-id="summary-of-differential-expression"><strong>Summary of differential expression</strong></h2>
<p>We inspect how many genes are significant at a false discovery rate (FDR) threshold of 0.05.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>de_summary <span class="ot">&lt;-</span> de_results <span class="sc">|&gt;</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_adj)) <span class="sc">|&gt;</span></span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_genes =</span> <span class="fu">n</span>(),</span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_significant =</span> <span class="fu">sum</span>(p_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">prop_significant =</span> n_significant <span class="sc">/</span> n_genes)</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>de_summary</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 3
  n_genes n_significant prop_significant
    &lt;int&gt;         &lt;int&gt;            &lt;dbl&gt;
1   14157             2         0.000141</code></pre>
</div>
</div>
<p>Only 2 out of 14,157 genes show a significant association with vital status at an FDR threshold of 0.05, representing ~0.014% of all tested genes. This indicates that differences between alive and dead patients are very subtle at the transcriptome-wide level, consistent with our PCA results, which also showed no clear separation between the two groups.</p>
</section>
<section id="volcano-plot" class="level2">
<h2 class="anchored" data-anchor-id="volcano-plot">Volcano plot</h2>
<p>We next create a volcano plot, showing the log2 fold change versus the negative log10 adjusted p-value. Genes with FDR &lt; 0.05 are highlighted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>volcano_data <span class="ot">&lt;-</span> de_results <span class="sc">|&gt;</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_adj)) <span class="sc">|&gt;</span></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neg_log10_p =</span> <span class="sc">-</span><span class="fu">log10</span>(p_adj),</span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">significant =</span> p_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>volcano_plot <span class="ot">&lt;-</span> volcano_data <span class="sc">|&gt;</span></span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2_fc, <span class="at">y =</span> neg_log10_p, <span class="at">colour =</span> significant)) <span class="sc">+</span></span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Differential expression: alive vs dead"</span>,</span>
<span id="cb125-10"><a href="#cb125-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"log2 fold change (alive - dead)"</span>,</span>
<span id="cb125-11"><a href="#cb125-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"-log10 adjusted p-value"</span>,</span>
<span id="cb125-12"><a href="#cb125-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">colour =</span> <span class="st">"FDR &lt; 0.05"</span>) <span class="sc">+</span></span>
<span id="cb125-13"><a href="#cb125-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb125-14"><a href="#cb125-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-15"><a href="#cb125-15" aria-hidden="true" tabindex="-1"></a>volcano_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-133-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"../results/07_volcano_alive_vs_dead.png"</span>,</span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> volcano_plot,</span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">height   =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The volcano plot shows that only a very small number of genes reach statistical significance after FDR correction (highlighted in blue). Most genes cluster near zero on the x-axis, indicating that log2 fold changes are extremely small between alive and dead patients. This suggests that vital status has only a weak effect on gene expression levels, consistent with our PCA results and the low number of significantly differentially expressed genes.</p>
</section>
<section id="top-differentially-expressed-genes" class="level2">
<h2 class="anchored" data-anchor-id="top-differentially-expressed-genes">Top differentially expressed genes</h2>
<p>We extract the top 20 genes with the smallest adjusted p-values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>top_de_genes <span class="ot">&lt;-</span> de_results <span class="sc">|&gt;</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_adj)) <span class="sc">|&gt;</span></span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p_adj) <span class="sc">|&gt;</span></span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a>top_de_genes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 6
   Ensembl_ID         mean_alive mean_dead log2_fc    p_value  p_adj
   &lt;chr&gt;                   &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;
 1 ENSG00000026751.17       1.19     0.914  0.271  0.00000592 0.0499
 2 ENSG00000225492.6        1.51     1.29   0.218  0.00000705 0.0499
 3 ENSG00000026950.17       2.14     2.03   0.105  0.0000110  0.0521
 4 ENSG00000168394.11       2.38     2.23   0.150  0.0000274  0.0647
 5 ENSG00000170476.16       1.28     0.972  0.311  0.0000254  0.0647
 6 ENSG00000198876.13       2.47     2.41   0.0597 0.0000188  0.0647
 7 ENSG00000156234.7        1.43     1.11   0.313  0.0000482  0.0949
 8 ENSG00000169248.13       1.87     1.60   0.273  0.0000603  0.0949
 9 ENSG00000234518.2        2.04     1.97   0.0685 0.0000570  0.0949
10 ENSG00000137168.8        2.56     2.49   0.0635 0.0000700  0.0961
11 ENSG00000162654.9        1.97     1.78   0.198  0.0000747  0.0961
12 ENSG00000090382.7        2.29     2.07   0.219  0.000126   0.113 
13 ENSG00000100902.11       1.17     1.08   0.0973 0.0000991  0.113 
14 ENSG00000137496.18       1.67     1.55   0.120  0.000128   0.113 
15 ENSG00000154451.14       1.26     1.04   0.219  0.000109   0.113 
16 ENSG00000240041.1        1.16     0.802  0.359  0.000117   0.113 
17 ENSG00000007341.19       1.33     1.26   0.0730 0.000252   0.128 
18 ENSG00000115415.20       2.56     2.48   0.0824 0.000239   0.128 
19 ENSG00000117228.10       2.40     2.28   0.121  0.000252   0.128 
20 ENSG00000120727.13       2.32     2.26   0.0603 0.000209   0.128 </code></pre>
</div>
</div>
<p>These genes show the strongest association between expression and vital status according to this simple model.</p>
</section>
<section id="boxplots-for-the-top-3-genes" class="level2">
<h2 class="anchored" data-anchor-id="boxplots-for-the-top-3-genes">Boxplots for the top 3 genes</h2>
<p>To visualise the effect more directly, we plot the distribution of log2 expression for the top 3 genes in alive vs dead patients.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>top3_ids <span class="ot">&lt;-</span> top_de_genes <span class="sc">|&gt;</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID)</span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>boxplot_data <span class="ot">&lt;-</span> full_data_alive <span class="sc">|&gt;</span></span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> top3_ids) <span class="sc">|&gt;</span></span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Ensembl_ID =</span> <span class="fu">factor</span>(Ensembl_ID, <span class="at">levels =</span> top3_ids))</span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>gene_boxplot <span class="ot">&lt;-</span> boxplot_data <span class="sc">|&gt;</span></span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> alive, <span class="at">y =</span> log2_expression, <span class="at">fill =</span> alive)) <span class="sc">+</span></span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb129-12"><a href="#cb129-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Ensembl_ID, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb129-13"><a href="#cb129-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb129-14"><a href="#cb129-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Top 3 differentially expressed genes"</span>,</span>
<span id="cb129-15"><a href="#cb129-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Alive status"</span>,</span>
<span id="cb129-16"><a href="#cb129-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"log2(expression + 1)"</span>) <span class="sc">+</span></span>
<span id="cb129-17"><a href="#cb129-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb129-18"><a href="#cb129-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-19"><a href="#cb129-19" aria-hidden="true" tabindex="-1"></a>gene_boxplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-136-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"../results/07_top3_boxplots_alive_vs_dead.png"</span>,</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> gene_boxplot,</span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For all three top differentially expressed genes, expression levels tend to be slightly higher in patients who are alive compared to those who are dead. However, the differences remain small in magnitude, consistent with the modest log2 fold changes observed in the volcano plot. While these genes show statistically significant associations with vital status, the effect sizes are limited, suggesting that the biological relevance may be subtle and that survival differences are unlikely to be driven by strong single-gene expression shifts.</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>