<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>02 - clean</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="02_clean_files/libs/clipboard/clipboard.min.js"></script>
<script src="02_clean_files/libs/quarto-html/quarto.js"></script>
<script src="02_clean_files/libs/quarto-html/popper.min.js"></script>
<script src="02_clean_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="02_clean_files/libs/quarto-html/anchor.min.js"></script>
<link href="02_clean_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="02_clean_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="02_clean_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="02_clean_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="02_clean_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">02 - clean</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="load-libraries" class="level1">
<h1>Load libraries</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="clean-gene-table" class="level1">
<h1>Clean gene table</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp.rds"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 60660   430</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="sc">|&gt;</span> </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 430
   Ensembl_ID         `TCGA-24-1104-01A` `TCGA-25-1320-01A` `TCGA-24-1850-01A`
   &lt;chr&gt;                           &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000267580.1               0                 0                   0    
 2 ENSG00000150628.7               0.152             0.263               0    
 3 ENSG00000207778.3               0                 0                   0    
 4 ENSG00000065526.12              3.81              3.45                2.52 
 5 ENSG00000179163.12              5.64              4.59                4.77 
 6 ENSG00000242248.2               0                 0.0771              0.272
 7 ENSG00000268020.3               0                 0                   0    
 8 ENSG00000267774.2               0                 0.0191              0    
 9 ENSG00000276627.3               0                 0.0106              0    
10 ENSG00000252727.1               0                 0                   0    
# ℹ 426 more variables: `TCGA-61-2101-01A` &lt;dbl&gt;, `TCGA-23-1111-01A` &lt;dbl&gt;,
#   `TCGA-25-2396-01A` &lt;dbl&gt;, `TCGA-30-1714-01A` &lt;dbl&gt;,
#   `TCGA-04-1347-01A` &lt;dbl&gt;, `TCGA-59-2351-01A` &lt;dbl&gt;,
#   `TCGA-09-1659-01B` &lt;dbl&gt;, `TCGA-24-1431-01A` &lt;dbl&gt;,
#   `TCGA-31-1951-01A` &lt;dbl&gt;, `TCGA-29-1769-01A` &lt;dbl&gt;,
#   `TCGA-13-0916-01A` &lt;dbl&gt;, `TCGA-61-1721-01A` &lt;dbl&gt;,
#   `TCGA-61-2008-02A` &lt;dbl&gt;, `TCGA-20-1687-01A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>We have 60600 observations (rows) and 430 variables (columns). The rows are the different genes and the columns are the samples. We have 60600 genes (that is too much) and we can see that some of them are 0 for all the samples. So we want to delete all the gene with no expression for each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="sc">-</span>Ensembl_ID)) <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 57825   430</code></pre>
</div>
</div>
<p>We steel have too much genes (57825) so we want to see which genes are unuseful</p>
<p>We calculate the sum of all the gene expression for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_gene_exp =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="sc">-</span>Ensembl_ID))) <span class="sc">|&gt;</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sum_gene_exp, <span class="at">.after =</span> Ensembl_ID) <span class="sc">|&gt;</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sum_gene_exp)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="sc">|&gt;</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 431
   Ensembl_ID         sum_gene_exp `TCGA-24-1104-01A` `TCGA-25-1320-01A`
   &lt;chr&gt;                     &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000260285.1         459.              1.19               1.11  
 2 ENSG00000224479.6          35.3             0.0144             0.137 
 3 ENSG00000252031.1          14.8             0                  0     
 4 ENSG00000188668.4          34.0             0                  0.0783
 5 ENSG00000228802.1          26.0             0.116              0.0475
 6 ENSG00000271716.1          19.5             0.169              0.136 
 7 ENSG00000100379.17       1048.              1.53               3.01  
 8 ENSG00000117298.16       1774.              3.80               3.98  
 9 ENSG00000259223.2          70.8             0                  0     
10 ENSG00000168070.12        302.              0.617              0.290 
# ℹ 427 more variables: `TCGA-24-1850-01A` &lt;dbl&gt;, `TCGA-61-2101-01A` &lt;dbl&gt;,
#   `TCGA-23-1111-01A` &lt;dbl&gt;, `TCGA-25-2396-01A` &lt;dbl&gt;,
#   `TCGA-30-1714-01A` &lt;dbl&gt;, `TCGA-04-1347-01A` &lt;dbl&gt;,
#   `TCGA-59-2351-01A` &lt;dbl&gt;, `TCGA-09-1659-01B` &lt;dbl&gt;,
#   `TCGA-24-1431-01A` &lt;dbl&gt;, `TCGA-31-1951-01A` &lt;dbl&gt;,
#   `TCGA-29-1769-01A` &lt;dbl&gt;, `TCGA-13-0916-01A` &lt;dbl&gt;,
#   `TCGA-61-1721-01A` &lt;dbl&gt;, `TCGA-61-2008-02A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Now we want to visualise this new varaible to know what treshold we are going to use to select the relevant genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_clean_files/figure-html/unnamed-chunk-5-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_clean_files/figure-html/unnamed-chunk-5-2.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">y =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_clean_files/figure-html/unnamed-chunk-5-3.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can see that a lot of genes have the sum around 0 (with the boxplot). In the histogram, the binwidth is 100 and 4 bars are higher than the others and correspond to irelevant genes. So we can choose a treshold of 400. There are 431 samples so a threshold of 400 means that in average, apromimativelly all the genes expressions are higher than 1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_gene_exp <span class="sc">&gt;</span> <span class="dv">400</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 14157   431</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_clean_files/figure-html/unnamed-chunk-7-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Now, we have 14157 genes which is more appropriate for our case.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(sum_gene_exp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gen_exp_clean, <span class="st">"../data/gen_exp_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-phenotype-table" class="level1">
<h1>Cleaning Phenotype table</h1>
<section id="loading-table" class="level2">
<h2 class="anchored" data-anchor-id="loading-table">Loading table</h2>
<p>We start by loading the “phenotype” table, with the raw clinical data. The slice_sample allows to see 20 random samples and start evaluating which variables (columns) will be useful</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype.rds"</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 84
   sample           id            disease_type case_id submitter_id primary_site
   &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;        &lt;chr&gt;   &lt;chr&gt;        &lt;chr&gt;       
 1 TCGA-13-0723-01A ff530f28-0ec… Cystic, Muc… ff530f… TCGA-13-0723 Ovary       
 2 TCGA-36-2538-01A 3ea880f4-7a7… Cystic, Muc… 3ea880… TCGA-36-2538 Ovary       
 3 TCGA-61-2104-01A 05263959-c4f… Cystic, Muc… 052639… TCGA-61-2104 Ovary       
 4 TCGA-72-4236-01A c8e0008e-00e… Cystic, Muc… c8e000… TCGA-72-4236 Ovary       
 5 TCGA-24-1436-01A 6e84e89d-4f3… Cystic, Muc… 6e84e8… TCGA-24-1436 Ovary       
 6 TCGA-24-1553-01A 08c64000-d32… Cystic, Muc… 08c640… TCGA-24-1553 Ovary       
 7 TCGA-13-A5FU-01A 0f7ac6ca-780… Cystic, Muc… 0f7ac6… TCGA-13-A5FU Ovary       
 8 TCGA-04-1357-01A feda41d8-ca5… Cystic, Muc… feda41… TCGA-04-1357 Ovary       
 9 TCGA-13-1819-01A 82e96c6c-a88… Cystic, Muc… 82e96c… TCGA-13-1819 Ovary       
10 TCGA-24-1427-01A 02d9aa2e-b16… Cystic, Muc… 02d9aa… TCGA-24-1427 Ovary       
11 TCGA-29-1702-01A 0e1598d8-cbe… Cystic, Muc… 0e1598… TCGA-29-1702 Ovary       
12 TCGA-04-1655-01A f8a5547f-f1b… Cystic, Muc… f8a554… TCGA-04-1655 Ovary       
13 TCGA-23-1029-01B 4d71dd15-cd0… Cystic, Muc… 4d71dd… TCGA-23-1029 Ovary       
14 TCGA-57-1993-01A 15fc17cd-d07… Cystic, Muc… 15fc17… TCGA-57-1993 Ovary       
15 TCGA-61-1911-01A 93095c36-226… Cystic, Muc… 93095c… TCGA-61-1911 Ovary       
16 TCGA-61-1998-01A 565d06a1-364… Cystic, Muc… 565d06… TCGA-61-1998 Ovary       
17 TCGA-25-1315-01A 635f5335-b00… Cystic, Muc… 635f53… TCGA-25-1315 Ovary       
18 TCGA-04-1516-11B 1c86e4e5-ecc… Cystic, Muc… 1c86e4… TCGA-04-1516 Ovary       
19 TCGA-20-0990-01A c9524775-6b5… Cystic, Muc… c95247… TCGA-20-0990 Ovary       
20 TCGA-61-2000-01A 29949b65-291… Cystic, Muc… 29949b… TCGA-61-2000 Ovary       
# ℹ 78 more variables: alcohol_history.exposures &lt;chr&gt;, race.demographic &lt;chr&gt;,
#   gender.demographic &lt;chr&gt;, ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   primary_site.project &lt;chr&gt;, project_id.project &lt;chr&gt;,
#   disease_type.project &lt;chr&gt;, name.project &lt;chr&gt;, …</code></pre>
</div>
</div>
<section id="removal-of-variables" class="level3">
<h3 class="anchored" data-anchor-id="removal-of-variables">Removal of variables</h3>
<p>We start by removing some unuseful variables, such as:</p>
<ul>
<li><p>“id”, “case_id”, “tissue_source_site_id.tissue_source_site”, “entity_id.annotations”, “annotation_id.annotations” and other:<br>
They are long lists of seemingly random characters, useful for identification of the sample but not for analysis, as we have no information on their meaning.<br>
We’ll use “sample” as identification.</p></li>
<li><p>“submitter_id”, “entity_submitter_id.annotations” :<br>
They are the same as “sample” without the last 3 characters. This is repetitive and untidy.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows how some variables repeat the same information </span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(sample, submitter_id)) <span class="sc">|&gt;</span> </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   sample           submitter_id
   &lt;chr&gt;            &lt;chr&gt;       
 1 TCGA-09-1664-01A TCGA-09-1664
 2 TCGA-61-1907-01A TCGA-61-1907
 3 TCGA-24-1846-01A TCGA-24-1846
 4 TCGA-29-1763-01A TCGA-29-1763
 5 TCGA-29-1766-01A TCGA-29-1766
 6 TCGA-13-0901-01A TCGA-13-0901
 7 TCGA-30-1861-01A TCGA-30-1861
 8 TCGA-23-1027-01A TCGA-23-1027
 9 TCGA-13-0893-01B TCGA-13-0893
10 TCGA-04-1519-01A TCGA-04-1519
11 TCGA-13-0801-01A TCGA-13-0801
12 TCGA-04-1338-01A TCGA-04-1338
13 TCGA-04-1341-01A TCGA-04-1341
14 TCGA-23-2641-01A TCGA-23-2641
15 TCGA-04-1347-01A TCGA-04-1347
16 TCGA-24-2019-01A TCGA-24-2019
17 TCGA-04-1362-01A TCGA-04-1362
18 TCGA-09-2051-01A TCGA-09-2051
19 TCGA-29-1690-01A TCGA-29-1690
20 TCGA-57-1584-01A TCGA-57-1584</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows identifier variables, no meaning for analysis </span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(id, case_id)) <span class="sc">|&gt;</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   id                                   case_id                             
   &lt;chr&gt;                                &lt;chr&gt;                               
 1 8c47ae58-91c7-4247-b420-50f31f5eb00d 8c47ae58-91c7-4247-b420-50f31f5eb00d
 2 a2319490-b85d-4219-a1b0-fa1ec432d5c8 a2319490-b85d-4219-a1b0-fa1ec432d5c8
 3 6746533a-8d0b-4ebc-87ec-49c8738121a8 6746533a-8d0b-4ebc-87ec-49c8738121a8
 4 ef5e85cf-2d82-4d52-a02c-fed5c76e4cec ef5e85cf-2d82-4d52-a02c-fed5c76e4cec
 5 dc9b4616-e815-43c8-9ec4-720a32868541 dc9b4616-e815-43c8-9ec4-720a32868541
 6 a09f0626-002a-48fd-8b2d-c66b8285cf23 a09f0626-002a-48fd-8b2d-c66b8285cf23
 7 2dfd8a0e-e642-49b3-abd7-f0334927eebb 2dfd8a0e-e642-49b3-abd7-f0334927eebb
 8 13319c20-02f6-4b5f-b24f-3d8f4084094c 13319c20-02f6-4b5f-b24f-3d8f4084094c
 9 23771dab-2aab-4d2b-95e2-9af7092cee6b 23771dab-2aab-4d2b-95e2-9af7092cee6b
10 ea6405c4-0fe4-4f37-bef2-275072f91e39 ea6405c4-0fe4-4f37-bef2-275072f91e39
11 6f592f17-e00b-4d04-a45b-d9d4cf4c3075 6f592f17-e00b-4d04-a45b-d9d4cf4c3075
12 0a48873a-092a-4b25-822b-b0b3c18c08a8 0a48873a-092a-4b25-822b-b0b3c18c08a8
13 0dd95368-0713-45f0-84f4-c91af9cbba74 0dd95368-0713-45f0-84f4-c91af9cbba74
14 33d36b64-4688-40a7-9a4f-7cd3d4cc683d 33d36b64-4688-40a7-9a4f-7cd3d4cc683d
15 dce71741-ccbe-40b7-a0b8-2048d07187a4 dce71741-ccbe-40b7-a0b8-2048d07187a4
16 56c4bd48-ab15-4b49-ae2f-968653052b50 56c4bd48-ab15-4b49-ae2f-968653052b50
17 4ca9c0e8-75a7-4665-9d88-6dd2bb44b1e1 4ca9c0e8-75a7-4665-9d88-6dd2bb44b1e1
18 914d8613-3c25-4a10-be50-28b42f4d3a5d 914d8613-3c25-4a10-be50-28b42f4d3a5d
19 5495b18a-10ec-427b-83b8-03ef4b587b13 5495b18a-10ec-427b-83b8-03ef4b587b13
20 82093ed9-a3c8-4e34-931f-4ec7ae745711 82093ed9-a3c8-4e34-931f-4ec7ae745711</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these variables </span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype <span class="sc">|&gt;</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(id, </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>            case_id, </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>            submitter_id, </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>            tissue_source_site_id.tissue_source_site, </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>            entity_submitter_id.annotations, </span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>            entity_id.annotations, </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>            annotation_id.annotations, </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>            case_id.annotations, </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>            treatment_id.treatments.diagnoses, </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>            submitter_id.treatments.diagnoses, </span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>            sample_id.samples, </span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>            pathology_report_uuid.samples)</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>         )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>“primary_site”, “primary_site.project”, “project_id.project”, … : These columns contain the same observation for each row, and will not be useful for analysis either.<br>
For example “gender.demographic” will always be “female”, as we are studying ovarian cancer.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows variables with repeated observations </span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(primary_site, </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>           name.program.project, </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>           last_known_disease_status.diagnoses, </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>           gender.demographic)) <span class="sc">|&gt;</span> </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 4
   primary_site name.program.project last_known_disease_sta…¹ gender.demographic
   &lt;chr&gt;        &lt;chr&gt;                &lt;chr&gt;                    &lt;chr&gt;             
 1 Ovary        TCGA                 not reported             female            
 2 Ovary        TCGA                 &lt;NA&gt;                     &lt;NA&gt;              
 3 Ovary        TCGA                 not reported             female            
 4 Ovary        TCGA                 not reported             female            
 5 Ovary        TCGA                 &lt;NA&gt;                     &lt;NA&gt;              
 6 Ovary        TCGA                 not reported             female            
 7 Ovary        TCGA                 not reported             female            
 8 Ovary        TCGA                 not reported             female            
 9 Ovary        TCGA                 not reported             female            
10 Ovary        TCGA                 not reported             female            
11 Ovary        TCGA                 not reported             female            
12 Ovary        TCGA                 &lt;NA&gt;                     &lt;NA&gt;              
13 Ovary        TCGA                 not reported             female            
14 Ovary        TCGA                 not reported             female            
15 Ovary        TCGA                 not reported             female            
16 Ovary        TCGA                 not reported             female            
17 Ovary        TCGA                 not reported             female            
18 Ovary        TCGA                 not reported             female            
19 Ovary        TCGA                 not reported             female            
20 Ovary        TCGA                 not reported             female            
# ℹ abbreviated name: ¹​last_known_disease_status.diagnoses</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these variables </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(primary_site, </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>            primary_site.project, </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>            project_id.project, </span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>            disease_type.project, </span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>            name.project, </span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>            name.program.project, </span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>            project.tissue_source_site, </span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>            days_to_diagnosis.diagnoses, </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>            last_known_disease_status.diagnoses, </span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>            tissue_or_organ_of_origin.diagnoses, </span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>            preservation_method.samples, </span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>            is_ffpe.samples, </span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>            gender.demographic)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>         )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>“entity_type.annotations” , “state.annotations”, … : These are empty or always the same when not empty. Samples might come from different sources, which would explain irregular conventions. After careful reading of the table, we can say that the fact that some observations are left empty has no other meaning.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows variables with repeated or empty observations </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(entity_type.annotations, </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>           state.annotations, </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>           status.annotations)) <span class="sc">|&gt;</span> </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 3
   entity_type.annotations state.annotations status.annotations
   &lt;chr&gt;                   &lt;chr&gt;             &lt;chr&gt;             
 1 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 2 case                    released          Approved          
 3 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 4 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 5 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 6 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 7 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 8 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
 9 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
10 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
11 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
12 case                    released          Approved          
13 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
14 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
15 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
16 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
17 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
18 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
19 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              
20 &lt;NA&gt;                    &lt;NA&gt;              &lt;NA&gt;              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these columns </span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(entity_type.annotations, </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>            state.annotations, </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>            status.annotations, </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>            composition.samples, </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>            days_to_sample_procurement.samples, </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>            state.treatments.diagnoses,</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>            tumor_grade.diagnoses,</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>            progression_or_recurrence.diagnoses))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The samples with a disease_type value other than “Cystic, Mucinous and Serous Neoplasms” miss a lot of other information (NA in other columns). We will not keep them as there are only 8 samples and will disturb analysis without providing new information. It now makes no sense to keep this column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check for other disease types </span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(disease_type <span class="sc">!=</span> <span class="st">"Cystic, Mucinous and Serous Neoplasms"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 51
  sample           disease_type alcohol_history.exposures race.demographic
  &lt;chr&gt;            &lt;chr&gt;        &lt;chr&gt;                     &lt;chr&gt;           
1 TCGA-01-0633-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
2 TCGA-01-0628-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
3 TCGA-01-0630-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
4 TCGA-01-0631-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
5 TCGA-01-0637-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
6 TCGA-01-0636-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
7 TCGA-01-0642-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
8 TCGA-01-0639-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
# ℹ 47 more variables: ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove samples with missing values </span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(disease_type <span class="sc">==</span> <span class="st">"Cystic, Mucinous and Serous Neoplasms"</span>)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove this column</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>disease_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The column prior_treatment.diagnoses is Not Reported or No for every samples except one Yes. This column will not be useful.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show values other than "Not Reported" and "No" for prior_treatment.diagnoses</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(prior_treatment.diagnoses<span class="sc">!=</span><span class="st">"Not Reported"</span> <span class="sc">&amp;</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>         prior_treatment.diagnoses <span class="sc">!=</span> <span class="st">"No"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(prior_treatment.diagnoses, <span class="at">.before =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 50
  prior_treatment.diagnoses sample       alcohol_history.expo…¹ race.demographic
  &lt;chr&gt;                     &lt;chr&gt;        &lt;chr&gt;                  &lt;chr&gt;           
1 Yes                       TCGA-61-172… Not Reported           white           
# ℹ abbreviated name: ¹​alcohol_history.exposures
# ℹ 46 more variables: ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>Some variables don’t always contain the same information on paper, but further analysis shows us that the observations are very similar. For example alcohol_history.exposures is “Not reported” or NA:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show NAs in alcohol_history.exposures</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(alcohol_history.exposures)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11 × 50
   sample          alcohol_history.expo…¹ race.demographic ethnicity.demographic
   &lt;chr&gt;           &lt;chr&gt;                  &lt;chr&gt;            &lt;chr&gt;                
 1 TCGA-36-2539-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 2 TCGA-72-4236-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 3 TCGA-72-4237-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 4 TCGA-72-4238-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 5 TCGA-72-4235-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 6 TCGA-72-4232-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 7 TCGA-72-4241-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 8 TCGA-72-4233-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 9 TCGA-72-4240-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
10 TCGA-72-4231-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
11 TCGA-72-4234-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
# ℹ abbreviated name: ¹​alcohol_history.exposures
# ℹ 46 more variables: vital_status.demographic &lt;chr&gt;,
#   age_at_index.demographic &lt;dbl&gt;, days_to_birth.demographic &lt;dbl&gt;,
#   year_of_birth.demographic &lt;dbl&gt;, days_to_death.demographic &lt;dbl&gt;,
#   year_of_death.demographic &lt;dbl&gt;, code.tissue_source_site &lt;chr&gt;,
#   name.tissue_source_site &lt;chr&gt;, bcr_id.tissue_source_site &lt;chr&gt;,
#   notes.annotations &lt;chr&gt;, submitter_id.annotations &lt;chr&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove this column</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>alcohol_history.exposures) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Modification of the variable vital_status.demographic to a binary variable “alive”</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">vital_status.demographic =</span> <span class="fu">case_when</span>(</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    vital_status.demographic <span class="sc">==</span> <span class="st">"Alive"</span> <span class="sc">~</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    vital_status.demographic <span class="sc">==</span> <span class="st">"Dead"</span> <span class="sc">~</span> <span class="cn">FALSE</span>,</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  ))<span class="sc">|&gt;</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">alive=</span>vital_status.demographic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(phenotype_clean, <span class="st">"../data/phenotype_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>