<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Full Analysis Pipeline</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00_all_files/libs/clipboard/clipboard.min.js"></script>
<script src="00_all_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="00_all_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="00_all_files/libs/quarto-html/popper.min.js"></script>
<script src="00_all_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00_all_files/libs/quarto-html/anchor.min.js"></script>
<link href="00_all_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00_all_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00_all_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00_all_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00_all_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Full Analysis Pipeline</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Download files from websource</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readr"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Download files only if missing</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">"../data/_raw/TCGA-OV.star_fpkm.tsv.gz"</span>)) {</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">"../data/_raw"</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  urls <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-OV.clinical.tsv.gz"</span>,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-OV.star_fpkm.tsv.gz"</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"../data/_raw"</span>, <span class="fu">basename</span>(urls))</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(urls)) {</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"Downloading "</span>, <span class="fu">basename</span>(urls[i]), <span class="st">" ..."</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">download.file</span>(urls[i], files[i])</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/_raw/TCGA-OV.star_fpkm.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 3950 Columns: 430
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr   (1): Ensembl_ID
dbl (429): TCGA-24-1104-01A, TCGA-25-1320-01A, TCGA-24-1850-01A, TCGA-61-210...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/_raw/TCGA-OV.clinical.tsv.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 632 Columns: 84
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (65): sample, id, disease_type, case_id, submitter_id, primary_site, alc...
dbl (17): age_at_index.demographic, days_to_birth.demographic, year_of_birth...
lgl  (2): oct_embedded.samples, is_ffpe.samples

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save lighter versions for later steps</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(phenotype, <span class="st">"../data/phenotype.rds"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gen_exp, <span class="st">"../data/gen_exp.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="load-libraries" class="level1">
<h1>Load libraries</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="clean-gene-table" class="level1">
<h1>Clean gene table</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp.rds"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3950  430</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="sc">|&gt;</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 430
   Ensembl_ID         `TCGA-24-1104-01A` `TCGA-25-1320-01A` `TCGA-24-1850-01A`
   &lt;chr&gt;                           &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000008394.13              4.87               2.30               4.14 
 2 ENSG00000110696.10              3.86               3.91               4.37 
 3 ENSG00000105963.15              0.851              0.763              0.890
 4 ENSG00000097007.19              3.49               4.15               4.71 
 5 ENSG00000087053.20              1.83               1.91               2.10 
 6 ENSG00000095397.16              1.63               2.02               1.99 
 7 ENSG00000068001.14              3.44               3.96               3.62 
 8 ENSG00000056586.16              2.76               2.61               2.12 
 9 ENSG00000070614.15              4.47               3.77               3.51 
10 ENSG00000109099.16              1.55               2.78               2.42 
# ℹ 426 more variables: `TCGA-61-2101-01A` &lt;dbl&gt;, `TCGA-23-1111-01A` &lt;dbl&gt;,
#   `TCGA-25-2396-01A` &lt;dbl&gt;, `TCGA-30-1714-01A` &lt;dbl&gt;,
#   `TCGA-04-1347-01A` &lt;dbl&gt;, `TCGA-59-2351-01A` &lt;dbl&gt;,
#   `TCGA-09-1659-01B` &lt;dbl&gt;, `TCGA-24-1431-01A` &lt;dbl&gt;,
#   `TCGA-31-1951-01A` &lt;dbl&gt;, `TCGA-29-1769-01A` &lt;dbl&gt;,
#   `TCGA-13-0916-01A` &lt;dbl&gt;, `TCGA-61-1721-01A` &lt;dbl&gt;,
#   `TCGA-61-2008-02A` &lt;dbl&gt;, `TCGA-20-1687-01A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>We have 60600 observations (rows) and 430 variables (columns). The rows are the different genes and the columns are the samples. We have 60600 genes (that is too much) and we can see that some of them are 0 for all the samples. So we want to delete all the gene with no expression for each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="sc">-</span>Ensembl_ID)) <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3943  430</code></pre>
</div>
</div>
<p>We still have too much genes (57825) so we want to see which genes are useless</p>
<p>We calculate the sum of all the gene expression for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sum_gene_exp =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="sc">-</span>Ensembl_ID))) <span class="sc">|&gt;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sum_gene_exp, <span class="at">.after =</span> Ensembl_ID) <span class="sc">|&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sum_gene_exp)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="sc">|&gt;</span> </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 431
   Ensembl_ID         sum_gene_exp `TCGA-24-1104-01A` `TCGA-25-1320-01A`
   &lt;chr&gt;                     &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000101464.11       1855.               3.31               4.13 
 2 ENSG00000082014.17       1641.               4.31               4.31 
 3 ENSG00000110243.12         10.9              0                  0    
 4 ENSG00000100376.12        683.               2.69               1.30 
 5 ENSG00000084444.14        441.               0.804              0.760
 6 ENSG00000074964.17        894.               2.69               2.26 
 7 ENSG00000099625.13        281.               0.354              0.375
 8 ENSG00000059122.17        941.               2.13               2.11 
 9 ENSG00000021762.20        780.               1.83               2.43 
10 ENSG00000101182.15       2826.               6.41               5.78 
# ℹ 427 more variables: `TCGA-24-1850-01A` &lt;dbl&gt;, `TCGA-61-2101-01A` &lt;dbl&gt;,
#   `TCGA-23-1111-01A` &lt;dbl&gt;, `TCGA-25-2396-01A` &lt;dbl&gt;,
#   `TCGA-30-1714-01A` &lt;dbl&gt;, `TCGA-04-1347-01A` &lt;dbl&gt;,
#   `TCGA-59-2351-01A` &lt;dbl&gt;, `TCGA-09-1659-01B` &lt;dbl&gt;,
#   `TCGA-24-1431-01A` &lt;dbl&gt;, `TCGA-31-1951-01A` &lt;dbl&gt;,
#   `TCGA-29-1769-01A` &lt;dbl&gt;, `TCGA-13-0916-01A` &lt;dbl&gt;,
#   `TCGA-61-1721-01A` &lt;dbl&gt;, `TCGA-61-2008-02A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Now we want to visualize this new variable to know what threshold we are going to use to select the relevant genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">y =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-15-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can see that a lot of genes have the sum around 0 (with the boxplot). In the histogram, the binwidth is 100 and 4 bars are higher than the others and correspond to irrelevant genes. So we can choose a threshold of 400. There are 431 samples so a threshold of 400 means that in average, approximately all the genes expressions are higher than 1.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sum_gene_exp <span class="sc">&gt;</span> <span class="dv">400</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gen_exp_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3004  431</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gen_exp_clean, <span class="fu">aes</span>(<span class="at">x =</span> sum_gene_exp)) <span class="sc">+</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Now, we have 14157 genes which is more appropriate for our case.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(sum_gene_exp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gen_exp_clean, <span class="st">"../data/gen_exp_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cleaning-phenotype-table" class="level1">
<h1>Cleaning Phenotype table</h1>
<section id="loading-table" class="level2">
<h2 class="anchored" data-anchor-id="loading-table">Loading table</h2>
<p>We start by loading the “phenotype” table, with the raw clinical data. The slice_sample allows to see 20 random samples and start evaluating which variables (columns) will be useful</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype.rds"</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 84
   sample           id            disease_type case_id submitter_id primary_site
   &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;        &lt;chr&gt;   &lt;chr&gt;        &lt;chr&gt;       
 1 TCGA-61-2113-01A 71faa2c1-0d5… Cystic, Muc… 71faa2… TCGA-61-2113 Ovary       
 2 TCGA-09-2056-01B ea6405c4-0fe… Cystic, Muc… ea6405… TCGA-09-2056 Ovary       
 3 TCGA-01-0630-11A dc9b4616-e81… Not Reported dc9b46… TCGA-01-0630 Ovary       
 4 TCGA-10-0928-01A 29d26e7e-f4a… Cystic, Muc… 29d26e… TCGA-10-0928 Ovary       
 5 TCGA-29-1698-01A 18d21132-a79… Cystic, Muc… 18d211… TCGA-29-1698 Ovary       
 6 TCGA-09-1675-01B 0fc8777c-8f1… Cystic, Muc… 0fc877… TCGA-09-1675 Ovary       
 7 TCGA-72-4241-01A 7f7c6821-6c1… Cystic, Muc… 7f7c68… TCGA-72-4241 Ovary       
 8 TCGA-29-2427-01A 6f25001a-f89… Cystic, Muc… 6f2500… TCGA-29-2427 Ovary       
 9 TCGA-13-1492-01A 63c6a89b-b28… Cystic, Muc… 63c6a8… TCGA-13-1492 Ovary       
10 TCGA-25-1317-01A 78777a80-cc6… Cystic, Muc… 78777a… TCGA-25-1317 Ovary       
11 TCGA-13-1494-01A f50ee039-1da… Cystic, Muc… f50ee0… TCGA-13-1494 Ovary       
12 TCGA-13-1412-01A 44493c23-82e… Cystic, Muc… 44493c… TCGA-13-1412 Ovary       
13 TCGA-25-2397-01A 4c099644-047… Cystic, Muc… 4c0996… TCGA-25-2397 Ovary       
14 TCGA-25-1315-01A 635f5335-b00… Cystic, Muc… 635f53… TCGA-25-1315 Ovary       
15 TCGA-20-0996-01A d7f82e34-5b3… Cystic, Muc… d7f82e… TCGA-20-0996 Ovary       
16 TCGA-23-1022-01A d0673efd-331… Cystic, Muc… d0673e… TCGA-23-1022 Ovary       
17 TCGA-61-1911-01A 93095c36-226… Cystic, Muc… 93095c… TCGA-61-1911 Ovary       
18 TCGA-23-1111-01A 6b2a9f9a-fe9… Cystic, Muc… 6b2a9f… TCGA-23-1111 Ovary       
19 TCGA-13-1410-01A c3f9ee36-d34… Cystic, Muc… c3f9ee… TCGA-13-1410 Ovary       
20 TCGA-04-1331-01A 6d10d4ee-633… Cystic, Muc… 6d10d4… TCGA-04-1331 Ovary       
# ℹ 78 more variables: alcohol_history.exposures &lt;chr&gt;, race.demographic &lt;chr&gt;,
#   gender.demographic &lt;chr&gt;, ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   primary_site.project &lt;chr&gt;, project_id.project &lt;chr&gt;,
#   disease_type.project &lt;chr&gt;, name.project &lt;chr&gt;, …</code></pre>
</div>
</div>
<section id="removal-of-variables" class="level3">
<h3 class="anchored" data-anchor-id="removal-of-variables">Removal of variables</h3>
<p>We start by removing some unuseful variables, such as:</p>
<ul>
<li><p>“id”, “case_id”, “tissue_source_site_id.tissue_source_site”, “entity_id.annotations”, “annotation_id.annotations” and other:<br>
They are long lists of seemingly random characters, useful for identification of the sample but not for analysis, as we have no information on their meaning.<br>
We’ll use “sample” as identification.</p></li>
<li><p>“submitter_id”, “entity_submitter_id.annotations” :<br>
They are the same as “sample” without the last 3 characters. This is repetitive and untidy.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows how some variables repeat the same information </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(sample, submitter_id)) <span class="sc">|&gt;</span> </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   sample           submitter_id
   &lt;chr&gt;            &lt;chr&gt;       
 1 TCGA-61-1904-01A TCGA-61-1904
 2 TCGA-24-1105-01A TCGA-24-1105
 3 TCGA-25-1625-01A TCGA-25-1625
 4 TCGA-57-1992-01A TCGA-57-1992
 5 TCGA-10-0937-01A TCGA-10-0937
 6 TCGA-13-1511-01A TCGA-13-1511
 7 TCGA-30-1856-01A TCGA-30-1856
 8 TCGA-61-1906-01A TCGA-61-1906
 9 TCGA-29-1777-01A TCGA-29-1777
10 TCGA-29-1783-01A TCGA-29-1783
11 TCGA-61-2095-01A TCGA-61-2095
12 TCGA-29-1774-01A TCGA-29-1774
13 TCGA-09-1661-01B TCGA-09-1661
14 TCGA-13-1817-01A TCGA-13-1817
15 TCGA-24-1562-01A TCGA-24-1562
16 TCGA-13-1504-01A TCGA-13-1504
17 TCGA-42-2593-01A TCGA-42-2593
18 TCGA-25-1628-01A TCGA-25-1628
19 TCGA-29-1770-01A TCGA-29-1770
20 TCGA-13-1491-01A TCGA-13-1491</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows identifier variables, no meaning for analysis </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="sc">|&gt;</span> </span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(id, case_id)) <span class="sc">|&gt;</span> </span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   id                                   case_id                             
   &lt;chr&gt;                                &lt;chr&gt;                               
 1 eef62876-7d17-496a-8971-2c7008d77014 eef62876-7d17-496a-8971-2c7008d77014
 2 602e2934-223a-44db-8c33-06eb05d72f94 602e2934-223a-44db-8c33-06eb05d72f94
 3 25a0a9e6-4f5b-45d8-8f34-abfd31d5ff1b 25a0a9e6-4f5b-45d8-8f34-abfd31d5ff1b
 4 c8acee1b-6854-409b-86ab-bf69dbe22ab6 c8acee1b-6854-409b-86ab-bf69dbe22ab6
 5 a8c4a333-6b67-44c8-9000-516f8ccea788 a8c4a333-6b67-44c8-9000-516f8ccea788
 6 2ebb738c-4829-4720-931a-96c0fd117e2a 2ebb738c-4829-4720-931a-96c0fd117e2a
 7 ab3dbbbe-eed6-4a35-a505-1815225e86c9 ab3dbbbe-eed6-4a35-a505-1815225e86c9
 8 c7c3fe4a-327a-4b7e-a259-989b72971209 c7c3fe4a-327a-4b7e-a259-989b72971209
 9 b25b3379-df8f-4900-9bce-0f8df50f549b b25b3379-df8f-4900-9bce-0f8df50f549b
10 7a08efd0-f984-430e-a8eb-1881047214b6 7a08efd0-f984-430e-a8eb-1881047214b6
11 e5a329ec-e50e-4786-901d-b58474f8ea65 e5a329ec-e50e-4786-901d-b58474f8ea65
12 2cdd5154-ee5c-4d67-b2eb-947dfaa56be2 2cdd5154-ee5c-4d67-b2eb-947dfaa56be2
13 b0585c7a-7205-4dac-9d54-6da89674accb b0585c7a-7205-4dac-9d54-6da89674accb
14 84f5876d-82b0-45ff-8f3d-a10653683454 84f5876d-82b0-45ff-8f3d-a10653683454
15 16829472-5666-4fba-9bf9-360ec1689aa0 16829472-5666-4fba-9bf9-360ec1689aa0
16 3491d67a-4f83-4f67-8085-bbb9fe942be5 3491d67a-4f83-4f67-8085-bbb9fe942be5
17 2a3a6222-f477-4198-ab9e-cdf527c1a6cf 2a3a6222-f477-4198-ab9e-cdf527c1a6cf
18 40635bf3-d8ba-4833-b623-547e55e5d07e 40635bf3-d8ba-4833-b623-547e55e5d07e
19 f0ca8205-0754-4bd8-a6a8-31901040c5e7 f0ca8205-0754-4bd8-a6a8-31901040c5e7
20 b45c543a-40d9-423a-a7fa-98e4b325b487 b45c543a-40d9-423a-a7fa-98e4b325b487</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these variables </span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype <span class="sc">|&gt;</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(id, </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>            case_id, </span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>            submitter_id, </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>            tissue_source_site_id.tissue_source_site, </span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>            entity_submitter_id.annotations, </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>            entity_id.annotations, </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>            annotation_id.annotations, </span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>            case_id.annotations, </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>            treatment_id.treatments.diagnoses, </span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>            submitter_id.treatments.diagnoses, </span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>            sample_id.samples, </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>            pathology_report_uuid.samples)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>         )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>“primary_site”, “primary_site.project”, “project_id.project”, … : These columns contain the same observation for each row, and will not be useful for analysis either.<br>
For example “gender.demographic” will always be “female”, as we are studying ovarian cancer.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows variables with repeated observations </span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(primary_site, </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>           name.program.project, </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>           last_known_disease_status.diagnoses, </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>           gender.demographic)) <span class="sc">|&gt;</span> </span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 4
   primary_site name.program.project last_known_disease_sta…¹ gender.demographic
   &lt;chr&gt;        &lt;chr&gt;                &lt;chr&gt;                    &lt;chr&gt;             
 1 Ovary        TCGA                 not reported             female            
 2 Ovary        TCGA                 not reported             female            
 3 Ovary        TCGA                 not reported             female            
 4 Ovary        TCGA                 not reported             female            
 5 Ovary        TCGA                 not reported             female            
 6 Ovary        TCGA                 not reported             female            
 7 Ovary        TCGA                 &lt;NA&gt;                     &lt;NA&gt;              
 8 Ovary        TCGA                 not reported             female            
 9 Ovary        TCGA                 not reported             female            
10 Ovary        TCGA                 not reported             female            
11 Ovary        TCGA                 not reported             female            
12 Ovary        TCGA                 not reported             female            
13 Ovary        TCGA                 not reported             female            
14 Ovary        TCGA                 not reported             female            
15 Ovary        TCGA                 not reported             female            
16 Ovary        TCGA                 not reported             female            
17 Ovary        TCGA                 not reported             female            
18 Ovary        TCGA                 not reported             female            
19 Ovary        TCGA                 not reported             female            
20 Ovary        TCGA                 not reported             female            
# ℹ abbreviated name: ¹​last_known_disease_status.diagnoses</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these variables </span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(primary_site, </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>            primary_site.project, </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>            project_id.project, </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>            disease_type.project, </span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>            name.project, </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>            name.program.project, </span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>            project.tissue_source_site, </span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>            days_to_diagnosis.diagnoses, </span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>            last_known_disease_status.diagnoses, </span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>            tissue_or_organ_of_origin.diagnoses, </span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>            preservation_method.samples, </span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>            is_ffpe.samples, </span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>            gender.demographic)</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>         )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>“entity_type.annotations” , “state.annotations”, … : These are empty or always the same when not empty. Samples might come from different sources, which would explain irregular conventions. After careful reading of the table, we can say that the fact that some observations are left empty has no other meaning.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows variables with repeated or empty observations </span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(entity_type.annotations, </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>           state.annotations, </span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>           status.annotations)) <span class="sc">|&gt;</span> </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">20</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 3
   entity_type.annotations state.annotations        status.annotations      
   &lt;chr&gt;                   &lt;chr&gt;                    &lt;chr&gt;                   
 1 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
 2 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
 3 case                    released                 Approved                
 4 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
 5 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
 6 case                    released                 Approved                
 7 ['case', 'case']        ['released', 'released'] ['Approved', 'Approved']
 8 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
 9 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
10 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
11 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
12 case                    released                 Approved                
13 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
14 case                    released                 Approved                
15 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
16 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
17 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
18 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
19 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    
20 &lt;NA&gt;                    &lt;NA&gt;                     &lt;NA&gt;                    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove these columns </span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(entity_type.annotations, </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>            state.annotations, </span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>            status.annotations, </span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>            composition.samples, </span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>            days_to_sample_procurement.samples, </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>            state.treatments.diagnoses,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>            tumor_grade.diagnoses,</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>            progression_or_recurrence.diagnoses))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The samples with a disease_type value other than “Cystic, Mucinous and Serous Neoplasms” miss a lot of other information (NA in other columns). We will not keep them as there are only 8 samples and will disturb analysis without providing new information. It now makes no sense to keep this column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check for other disease types </span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(disease_type <span class="sc">!=</span> <span class="st">"Cystic, Mucinous and Serous Neoplasms"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 51
  sample           disease_type alcohol_history.exposures race.demographic
  &lt;chr&gt;            &lt;chr&gt;        &lt;chr&gt;                     &lt;chr&gt;           
1 TCGA-01-0633-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
2 TCGA-01-0628-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
3 TCGA-01-0630-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
4 TCGA-01-0631-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
5 TCGA-01-0637-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
6 TCGA-01-0636-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
7 TCGA-01-0642-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
8 TCGA-01-0639-11A Not Reported &lt;NA&gt;                      &lt;NA&gt;            
# ℹ 47 more variables: ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove samples with missing values </span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(disease_type <span class="sc">==</span> <span class="st">"Cystic, Mucinous and Serous Neoplasms"</span>)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove this column</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>disease_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The column prior_treatment.diagnoses is Not Reported or No for every samples except one Yes. This column will not be useful.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show values other than "Not Reported" and "No" for prior_treatment.diagnoses</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(prior_treatment.diagnoses<span class="sc">!=</span><span class="st">"Not Reported"</span> <span class="sc">&amp;</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>         prior_treatment.diagnoses <span class="sc">!=</span> <span class="st">"No"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(prior_treatment.diagnoses, <span class="at">.before =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 50
  prior_treatment.diagnoses sample       alcohol_history.expo…¹ race.demographic
  &lt;chr&gt;                     &lt;chr&gt;        &lt;chr&gt;                  &lt;chr&gt;           
1 Yes                       TCGA-61-172… Not Reported           white           
# ℹ abbreviated name: ¹​alcohol_history.exposures
# ℹ 46 more variables: ethnicity.demographic &lt;chr&gt;,
#   vital_status.demographic &lt;chr&gt;, age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>Some variables don’t always contain the same information on paper, but further analysis shows us that the observations are very similar. For example alcohol_history.exposures is “Not reported” or NA:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show NAs in alcohol_history.exposures</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(alcohol_history.exposures)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11 × 50
   sample          alcohol_history.expo…¹ race.demographic ethnicity.demographic
   &lt;chr&gt;           &lt;chr&gt;                  &lt;chr&gt;            &lt;chr&gt;                
 1 TCGA-36-2539-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 2 TCGA-72-4236-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 3 TCGA-72-4237-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 4 TCGA-72-4238-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 5 TCGA-72-4235-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 6 TCGA-72-4232-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 7 TCGA-72-4241-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 8 TCGA-72-4233-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
 9 TCGA-72-4240-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
10 TCGA-72-4231-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
11 TCGA-72-4234-0… &lt;NA&gt;                   &lt;NA&gt;             &lt;NA&gt;                 
# ℹ abbreviated name: ¹​alcohol_history.exposures
# ℹ 46 more variables: vital_status.demographic &lt;chr&gt;,
#   age_at_index.demographic &lt;dbl&gt;, days_to_birth.demographic &lt;dbl&gt;,
#   year_of_birth.demographic &lt;dbl&gt;, days_to_death.demographic &lt;dbl&gt;,
#   year_of_death.demographic &lt;dbl&gt;, code.tissue_source_site &lt;chr&gt;,
#   name.tissue_source_site &lt;chr&gt;, bcr_id.tissue_source_site &lt;chr&gt;,
#   notes.annotations &lt;chr&gt;, submitter_id.annotations &lt;chr&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove this column</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>alcohol_history.exposures) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Modification of the variable vital_status.demographic to a binary variable “alive”</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> phenotype_clean <span class="sc">|&gt;</span> </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">vital_status.demographic =</span> <span class="fu">case_when</span>(</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>    vital_status.demographic <span class="sc">==</span> <span class="st">"Alive"</span> <span class="sc">~</span> <span class="cn">TRUE</span>,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>    vital_status.demographic <span class="sc">==</span> <span class="st">"Dead"</span> <span class="sc">~</span> <span class="cn">FALSE</span>,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  ))<span class="sc">|&gt;</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">alive=</span>vital_status.demographic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(phenotype_clean, <span class="st">"../data/phenotype_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-libraries-1" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-1">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-cleaned-tables" class="level2">
<h2 class="anchored" data-anchor-id="load-cleaned-tables">Load cleaned tables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp_clean.rds"</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="reshape-gene-expression-table" class="level2">
<h2 class="anchored" data-anchor-id="reshape-gene-expression-table">Reshape gene expression table</h2>
<p>The current gene expression dataset is in a wide format:<br>
rows = genes columns = samples<br>
For further analyses, it is more convenient to work in a long format: one row per (gene × sample) measurement. The dataset is now tidy.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>gen_exp_long <span class="ot">&lt;-</span> gen_exp_clean <span class="sc">|&gt;</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span><span class="fu">c</span>(Ensembl_ID),</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expression"</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>               )</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>gen_exp_long <span class="sc">|&gt;</span> </span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 3
   Ensembl_ID         sample           expression
   &lt;chr&gt;              &lt;chr&gt;                 &lt;dbl&gt;
 1 ENSG00000101849.17 TCGA-04-1350-01A       1.70
 2 ENSG00000105321.14 TCGA-36-1580-01A       2.73
 3 ENSG00000106565.18 TCGA-13-0890-01A       6.91
 4 ENSG00000103043.15 TCGA-24-1430-01A       3.35
 5 ENSG00000077097.15 TCGA-24-1928-01A       4.71
 6 ENSG00000071794.16 TCGA-13-1510-01A       2.20
 7 ENSG00000100722.20 TCGA-13-0797-01A       2.00
 8 ENSG00000107554.17 TCGA-29-1774-01A       1.93
 9 ENSG00000054282.16 TCGA-24-1467-01A       2.38
10 ENSG00000025293.17 TCGA-24-2254-01A       2.45</code></pre>
</div>
</div>
</section>
<section id="join-clinical-metadata" class="level2">
<h2 class="anchored" data-anchor-id="join-clinical-metadata">Join clinical metadata</h2>
<p>Now we merge the gene expression table with the phenotype table using the common identifier sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge expression data with phenotype metadata</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> gen_exp_long <span class="sc">|&gt;</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(phenotype_clean, <span class="at">by =</span> <span class="st">"sample"</span>) <span class="sc">|&gt;</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sample, <span class="at">.before =</span> Ensembl_ID)</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>full_data <span class="sc">|&gt;</span> </span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 51
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-13-1… ENSG00000…       3.73 white            not hispanic or lati… FALSE
 2 TCGA-57-1… ENSG00000…       3.73 white            not hispanic or lati… TRUE 
 3 TCGA-04-1… ENSG00000…       1.66 white            not reported          FALSE
 4 TCGA-24-1… ENSG00000…       2.86 white            not reported          FALSE
 5 TCGA-24-1… ENSG00000…       1.54 white            not reported          TRUE 
 6 TCGA-36-1… ENSG00000…       2.77 white            not reported          FALSE
 7 TCGA-25-2… ENSG00000…       3.26 white            not hispanic or lati… FALSE
 8 TCGA-13-2… ENSG00000…       3.27 white            not reported          TRUE 
 9 TCGA-61-1… ENSG00000…       3.44 white            not hispanic or lati… TRUE 
10 TCGA-30-1… ENSG00000…       1.55 white            not hispanic or lati… FALSE
# ℹ 45 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>We now have: biological measurements (gene-exp) + patient characteristics (phenotype).</p>
</section>
<section id="create-additional-variables" class="level2">
<h2 class="anchored" data-anchor-id="create-additional-variables">Create additional variables</h2>
<p>We can also create new useful variables like:</p>
<ul>
<li><p>A log-transformed expression variable (log2_expression), commonly used in transcriptomics to reduce skewness and easier interpretation.</p></li>
<li><p>A categorical age variable (age_cat), based on common thresholds in clinical studies.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">&lt;-</span> full_data <span class="sc">|&gt;</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log2_expression =</span> <span class="fu">log2</span>(expression <span class="sc">+</span> <span class="dv">1</span>),</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">age_cat =</span> <span class="fu">case_when</span>(</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>           age_at_index.demographic <span class="sc">&lt;</span> <span class="dv">50</span> <span class="sc">~</span> <span class="st">"young"</span>,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>           age_at_index.demographic <span class="sc">&gt;=</span> <span class="dv">50</span> <span class="sc">&amp;</span> age_at_index.demographic <span class="sc">&lt;</span> <span class="dv">65</span> <span class="sc">~</span> <span class="st">"middle"</span>,</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>           age_at_index.demographic <span class="sc">&gt;=</span> <span class="dv">65</span> <span class="sc">~</span> <span class="st">"senior"</span>,</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>           <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_character_</span>)</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>         )</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>full_data <span class="sc">|&gt;</span> </span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-24-1… ENSG00000…       3.31 white            not reported          TRUE 
 2 TCGA-30-1… ENSG00000…       3.21 white            not hispanic or lati… FALSE
 3 TCGA-04-1… ENSG00000…       4.34 white            hispanic or latino    FALSE
 4 TCGA-09-2… ENSG00000…       2.03 white            not hispanic or lati… TRUE 
 5 TCGA-24-2… ENSG00000…       3.96 white            not reported          FALSE
 6 TCGA-61-2… ENSG00000…       1.96 white            not reported          TRUE 
 7 TCGA-13-0… ENSG00000…       2.74 white            not hispanic or lati… TRUE 
 8 TCGA-24-1… ENSG00000…       1.64 black or africa… not reported          FALSE
 9 TCGA-20-1… ENSG00000…       1.63 white            not hispanic or lati… TRUE 
10 TCGA-24-1… ENSG00000…       2.07 white            not reported          FALSE
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
<p>The +1 in the log transformation prevents issues when expression = 0. (goes to -inf with log)</p>
<p>NA_character_ ensures that missing age values remain coded as valid character-type missing values in the age category variable.</p>
</section>
<section id="save-augmented-dataset" class="level2">
<h2 class="anchored" data-anchor-id="save-augmented-dataset">Save augmented dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(full_data, <span class="st">"../data/data_augmented.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="loading-libraries-and-data" class="level3">
<h3 class="anchored" data-anchor-id="loading-libraries-and-data">0.Loading libraries and data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load 3 tables</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype_clean.rds"</span>)</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp_clean.rds"</span>)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="general-information" class="level2">
<h2 class="anchored" data-anchor-id="general-information">1. General information</h2>
<p>The analysis will be based on the article <em>Comparison of gene expression patterns across twelve tumor types identifies a cancer supercluster characterized by TP53 mutations and cell cycle defects</em>. The datasets used in the article are from the The Cancer Genome Atlas (TCGA) ovarian cancer collection. In the article, compares gene-expression profiles across multiple tumor types to identify transcriptional superclusters.</p>
<p>Two different data sets are used:</p>
<ul>
<li><p>A clinical dataset: TCGA-OV.clinical.tsv</p></li>
<li><p>A gene expression dataset: TCGA-OV.star_fpkm.tsv</p></li>
</ul>
<p>The clinical and transcriptomic datasets for the TCGA ovarian cancer cohort will be used to search for gene expression associations with clinical variables and outcomes.</p>
</section>
<section id="dataset-descriptions" class="level2">
<h2 class="anchored" data-anchor-id="dataset-descriptions">2. Dataset descriptions</h2>
<section id="clinical-dataset" class="level3">
<h3 class="anchored" data-anchor-id="clinical-dataset">Clinical dataset</h3>
<ul>
<li><p>Format: Tab-separated values (TSV) file.</p></li>
<li><p>Contains clinical variables from the TCGA ovarian cancer cohort. As it contains observations about the patients, it is called <code>phenotype</code> in this project.</p></li>
<li><p>Size: There are 632 patients as rows and 84 clinical variables as columns.</p></li>
<li><p>Key variables and data types: The following variables are most relevant for the analysis, - <code>alive</code> (bool),</p>
<ul>
<li><code>figo_stage.diagnoses</code> (chr),</li>
<li><code>morphology.diagnoses</code> (chr, the first part encodes the tumor type, the second part encodes the malignancy),</li>
<li><code>age_at_earliest_diagnosis_in_years.diagnoses.xena_derived</code> (numeric),</li>
<li><code>treatment_type.treatments.diagnoses</code> (chr, indicating the type of treatement),</li>
</ul></li>
<li><p>Needed cleaning: The dataset contains a lot of redundant variables (e.g.&nbsp;many instances of a variable indicating the ID), many have been combined into a single one or have been removed altogether. Also the variable containing information about the gender has been removed as all patients with ovarian cancer are female. Some variables contain annotations to the patients, but as most are empty and irrelevant for the analysis they have been removed as well.</p></li>
<li><p>Size after cleaning: 624 patients on the rows, 52 clinical variables on the columns.</p></li>
</ul>
<p>Underneath, 10 randomly selected rows of the set are shown to understand the structure:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>phenotype_clean <span class="sc">|&gt;</span> <span class="fu">slice_sample</span>(<span class="at">n=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 49
   sample    race.demographic ethnicity.demographic alive age_at_index.demogra…¹
   &lt;chr&gt;     &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;                  &lt;dbl&gt;
 1 TCGA-29-… white            not reported          TRUE                      72
 2 TCGA-13-… white            not hispanic or lati… FALSE                     53
 3 TCGA-13-… white            not hispanic or lati… TRUE                      51
 4 TCGA-13-… white            not hispanic or lati… FALSE                     73
 5 TCGA-13-… white            not hispanic or lati… FALSE                     66
 6 TCGA-13-… black or africa… not reported          FALSE                     60
 7 TCGA-24-… white            not reported          TRUE                      64
 8 TCGA-25-… white            not hispanic or lati… FALSE                     64
 9 TCGA-13-… white            not hispanic or lati… FALSE                     62
10 TCGA-09-… white            not hispanic or lati… TRUE                      57
# ℹ abbreviated name: ¹​age_at_index.demographic
# ℹ 44 more variables: days_to_birth.demographic &lt;dbl&gt;,
#   year_of_birth.demographic &lt;dbl&gt;, days_to_death.demographic &lt;dbl&gt;,
#   year_of_death.demographic &lt;dbl&gt;, code.tissue_source_site &lt;chr&gt;,
#   name.tissue_source_site &lt;chr&gt;, bcr_id.tissue_source_site &lt;chr&gt;,
#   notes.annotations &lt;chr&gt;, submitter_id.annotations &lt;chr&gt;,
#   classification.annotations &lt;chr&gt;, created_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
<section id="gene-expression-dataset" class="level3">
<h3 class="anchored" data-anchor-id="gene-expression-dataset">Gene expression dataset</h3>
<ul>
<li><p>Format: Tab-separated values (TSV) file.</p></li>
<li><p>Contains gene expression data (FPKM) from the TCGA ovarian cancer cohort.</p></li>
<li><p>The values are FPKM (Fragments Per Kilobase of transcript per Million mapped reads) which is a normalized gene-expression metric.</p></li>
<li><p>Size: There are 60660 genes on the rows and 430 samples on the columns.</p></li>
<li><p>Data types: numeric non‐negative values</p></li>
<li><p>Needed cleaning: Some genes have a null expression level for all samples, these genes have been removed from the dataset. To further reduce the number of genes and keep only the most relevant ones, the summed expression level over all samples for each gene has been computed and the relevant genes were kept based on a threshold.</p></li>
<li><p>Size after cleaning: 14157 genes on the rows, 431 variables on the columns.</p></li>
</ul>
<p>Underneath, 10 randomly selected rows of the set are shown to understand the structure:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>gen_exp_clean <span class="sc">|&gt;</span> <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 430
   Ensembl_ID         `TCGA-24-1104-01A` `TCGA-25-1320-01A` `TCGA-24-1850-01A`
   &lt;chr&gt;                           &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
 1 ENSG00000050393.11               3.70               3.07               3.32
 2 ENSG00000105245.9                3.49               2.86               3.07
 3 ENSG00000075391.17               1.03               1.86               1.43
 4 ENSG00000089009.15               6.91               7.34               7.28
 5 ENSG00000100650.16               4.64               5.06               4.58
 6 ENSG00000075131.10               2.74               1.50               1.63
 7 ENSG00000105983.22               2.60               2.48               3.16
 8 ENSG00000088766.12               1.17               1.39               2.14
 9 ENSG00000010626.15               2.62               1.33               2.13
10 ENSG00000103197.18               2.66               3.32               3.24
# ℹ 426 more variables: `TCGA-61-2101-01A` &lt;dbl&gt;, `TCGA-23-1111-01A` &lt;dbl&gt;,
#   `TCGA-25-2396-01A` &lt;dbl&gt;, `TCGA-30-1714-01A` &lt;dbl&gt;,
#   `TCGA-04-1347-01A` &lt;dbl&gt;, `TCGA-59-2351-01A` &lt;dbl&gt;,
#   `TCGA-09-1659-01B` &lt;dbl&gt;, `TCGA-24-1431-01A` &lt;dbl&gt;,
#   `TCGA-31-1951-01A` &lt;dbl&gt;, `TCGA-29-1769-01A` &lt;dbl&gt;,
#   `TCGA-13-0916-01A` &lt;dbl&gt;, `TCGA-61-1721-01A` &lt;dbl&gt;,
#   `TCGA-61-2008-02A` &lt;dbl&gt;, `TCGA-20-1687-01A` &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="joining-of-the-datasets" class="level3">
<h3 class="anchored" data-anchor-id="joining-of-the-datasets">Joining of the datasets</h3>
<p>These two datasets have been combined to one by matching the patient IDs (samples). As the two datasets do not show the same amount of samples (the phenotype dataset is bigger than the gene expression dataset), the IDs that could not be resolved between the two sets have been removed. This leaves only patients who have recorded gene expression data (no gene expression data is lost).</p>
<p>The new size is 6’073’353 rows and 53 columns.</p>
<p>Two variables have been added: - <code>log2_expression</code> (numeric, gene expression on logarithmic scale) - <code>age_cat</code> (chr, age category the patient belongs to: young, middle, senior)</p>
<p>In addition to the key variables mentioned for the phenotype (clinical) data and the gene expression values from the gene expression data, these two variables will allow for easier visualization and analysis.</p>
<p>In the <code>figo_stage.diagnoses</code> variable, the tumor stages IA, IB and the majority of IC have disappeared after joining. This is due to the fact that it it rare to have gene expression data for early stage cancers.</p>
<p>Underneath, the 10 first rows of the joined set are shown to understand the structure:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> <span class="fu">head</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-24-1… ENSG00000…      0.918 white            not reported          FALSE
 2 TCGA-25-1… ENSG00000…      1.29  white            not hispanic or lati… FALSE
 3 TCGA-24-1… ENSG00000…      0.522 white            not reported          TRUE 
 4 TCGA-61-2… ENSG00000…      1.62  white            not hispanic or lati… FALSE
 5 TCGA-23-1… ENSG00000…      0.658 white            not reported          TRUE 
 6 TCGA-25-2… ENSG00000…      0.374 white            not hispanic or lati… FALSE
 7 TCGA-30-1… ENSG00000…      0.515 white            not hispanic or lati… FALSE
 8 TCGA-04-1… ENSG00000…      0.694 white            not hispanic or lati… TRUE 
 9 TCGA-59-2… ENSG00000…      0.372 white            not hispanic or lati… TRUE 
10 TCGA-09-1… ENSG00000…      0.544 white            not hispanic or lati… FALSE
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
</section>
<section id="quality-and-intergrity" class="level2">
<h2 class="anchored" data-anchor-id="quality-and-intergrity">3. Quality and intergrity</h2>
<p>In the phenotype datasets, a few columns containing only NAs have been removed. Some of the variables contained a label “Not Reported” instead of NA. As described above, many variables described redundant information and needed to be removed.</p>
<p>The gene expression dataset is of good quality and integrity, the only correction needed was to remove uninformative genes.</p>
<p>Overall the data seems of good quality without aberrant values or errors in the writing of the sets.</p>
</section>
<section id="visualisation" class="level2">
<h2 class="anchored" data-anchor-id="visualisation">4. Visualisation</h2>
<p>The following graphs will allow to visualize the content of the data set.</p>
<p>A list of the variables contained in the full dataset:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> <span class="fu">glimpse</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 1,288,716
Columns: 53
$ sample                                                    &lt;chr&gt; "TCGA-24-110…
$ Ensembl_ID                                                &lt;chr&gt; "ENSG0000010…
$ expression                                                &lt;dbl&gt; 0.9183099, 1…
$ race.demographic                                          &lt;chr&gt; "white", "wh…
$ ethnicity.demographic                                     &lt;chr&gt; "not reporte…
$ alive                                                     &lt;lgl&gt; FALSE, FALSE…
$ age_at_index.demographic                                  &lt;dbl&gt; 56, 65, 72, …
$ days_to_birth.demographic                                 &lt;dbl&gt; -20527, -240…
$ year_of_birth.demographic                                 &lt;dbl&gt; 1945, 1936, …
$ days_to_death.demographic                                 &lt;dbl&gt; 1933, 1155, …
$ year_of_death.demographic                                 &lt;dbl&gt; 2006, 2004, …
$ code.tissue_source_site                                   &lt;chr&gt; "24", "25", …
$ name.tissue_source_site                                   &lt;chr&gt; "Washington …
$ bcr_id.tissue_source_site                                 &lt;chr&gt; "IGC", "IGC"…
$ notes.annotations                                         &lt;chr&gt; NA, NA, NA, …
$ submitter_id.annotations                                  &lt;chr&gt; NA, NA, NA, …
$ classification.annotations                                &lt;chr&gt; NA, NA, NA, …
$ created_datetime.annotations                              &lt;chr&gt; NA, NA, NA, …
$ updated_datetime.annotations                              &lt;chr&gt; NA, NA, NA, …
$ category.annotations                                      &lt;chr&gt; NA, NA, NA, …
$ case_submitter_id.annotations                             &lt;chr&gt; NA, NA, NA, …
$ figo_stage.diagnoses                                      &lt;chr&gt; "Stage IV", …
$ synchronous_malignancy.diagnoses                          &lt;chr&gt; "Not Reporte…
$ days_to_last_follow_up.diagnoses                          &lt;dbl&gt; 1933, 1124, …
$ age_at_diagnosis.diagnoses                                &lt;dbl&gt; 20527, 24018…
$ primary_diagnosis.diagnoses                               &lt;chr&gt; "Serous cyst…
$ prior_malignancy.diagnoses                                &lt;chr&gt; "not reporte…
$ year_of_diagnosis.diagnoses                               &lt;dbl&gt; 2001, 2001, …
$ prior_treatment.diagnoses                                 &lt;chr&gt; "No", "No", …
$ morphology.diagnoses                                      &lt;chr&gt; "8441/3", "8…
$ classification_of_tumor.diagnoses                         &lt;chr&gt; "not reporte…
$ icd_10_code.diagnoses                                     &lt;chr&gt; "C56.9", "C5…
$ site_of_resection_or_biopsy.diagnoses                     &lt;chr&gt; "Ovary", "Ov…
$ age_at_earliest_diagnosis.diagnoses.xena_derived          &lt;dbl&gt; 20527, 24018…
$ age_at_earliest_diagnosis_in_years.diagnoses.xena_derived &lt;dbl&gt; 56.23836, 65…
$ updated_datetime.treatments.diagnoses                     &lt;chr&gt; "['2019-07-3…
$ treatment_type.treatments.diagnoses                       &lt;chr&gt; "['Radiation…
$ treatment_or_therapy.treatments.diagnoses                 &lt;chr&gt; "['yes', 'ye…
$ created_datetime.treatments.diagnoses                     &lt;chr&gt; "['', '2019-…
$ sample_type_id.samples                                    &lt;chr&gt; "01", "01", …
$ tumor_descriptor.samples                                  &lt;chr&gt; "Primary", "…
$ sample_type.samples                                       &lt;chr&gt; "Primary Tum…
$ days_to_collection.samples                                &lt;dbl&gt; NA, NA, NA, …
$ initial_weight.samples                                    &lt;dbl&gt; NA, NA, NA, …
$ intermediate_dimension.samples                            &lt;dbl&gt; 1.0, 3.0, 1.…
$ shortest_dimension.samples                                &lt;dbl&gt; 0.7, 1.0, 0.…
$ oct_embedded.samples                                      &lt;lgl&gt; NA, NA, NA, …
$ specimen_type.samples                                     &lt;chr&gt; "Unknown", "…
$ longest_dimension.samples                                 &lt;dbl&gt; 1.5, 3.4, 2.…
$ tissue_type.samples                                       &lt;chr&gt; "Tumor", "Tu…
$ annotations.samples                                       &lt;chr&gt; NA, NA, NA, …
$ log2_expression                                           &lt;dbl&gt; 0.9398358, 1…
$ age_cat                                                   &lt;chr&gt; "middle", "s…</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>data_augmented_numeric_summary <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> </span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(),</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">list</span>(<span class="at">mean =</span> <span class="sc">~</span><span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">median =</span> <span class="sc">~</span><span class="fu">median</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sd =</span> <span class="sc">~</span><span class="fu">sd</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min =</span> <span class="sc">~</span><span class="fu">min</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">max =</span> <span class="sc">~</span><span class="fu">max</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">missing =</span> <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.x))),</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">.names =</span> <span class="st">"{.col}_{.fn}"</span>)) </span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>data_augmented_numeric_summary</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 102
  expression_mean expression_median expression_sd expression_min expression_max
            &lt;dbl&gt;             &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;          &lt;dbl&gt;
1            3.03              2.83          1.60              0           13.7
# ℹ 97 more variables: expression_missing &lt;int&gt;,
#   age_at_index.demographic_mean &lt;dbl&gt;, age_at_index.demographic_median &lt;dbl&gt;,
#   age_at_index.demographic_sd &lt;dbl&gt;, age_at_index.demographic_min &lt;dbl&gt;,
#   age_at_index.demographic_max &lt;dbl&gt;, age_at_index.demographic_missing &lt;int&gt;,
#   days_to_birth.demographic_mean &lt;dbl&gt;,
#   days_to_birth.demographic_median &lt;dbl&gt;, days_to_birth.demographic_sd &lt;dbl&gt;,
#   days_to_birth.demographic_min &lt;dbl&gt;, days_to_birth.demographic_max &lt;dbl&gt;, …</code></pre>
</div>
</div>
<section id="distribution-of-the-age-at-diagnosis" class="level3">
<h3 class="anchored" data-anchor-id="distribution-of-the-age-at-diagnosis">Distribution of the age at diagnosis</h3>
<p>The graph is a histogram showing the distribution of patient age at diagnosis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>patient_level <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>    sample,</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">age =</span> age_at_earliest_diagnosis_in_years.diagnoses.xena_derived</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age))</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>patient_level <span class="sc">|&gt;</span> </span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> age)) <span class="sc">+</span></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">"skyblue"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">mean</span>(age)), <span class="at">color =</span> <span class="st">"red"</span>,</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Distribution of Age at Diagnosis (Patient Level)"</span>,</span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Age"</span>,</span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Number of Patients"</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-66-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/04_age_distribution.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<p>The age of patients at diagnosis seems to be normally distributed with a mean (red line) at around ~60 years.</p>
</section>
<section id="age-distribution-at-tumor-stage" class="level3">
<h3 class="anchored" data-anchor-id="age-distribution-at-tumor-stage">Age distribution at tumor stage</h3>
<p>Tumor stages IA, IB and the majority of IC have disappeared after joining the initial two datasets. Most patients, out of those which also have gene expression data, have ovarian cancer stage IIIC. There are very few data points for stages IC, IIA, IIB and IIIA.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sample, figo_stage.diagnoses) <span class="sc">|&gt;</span> </span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(figo_stage.diagnoses)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 9 × 2
  figo_stage.diagnoses     n
  &lt;chr&gt;                &lt;int&gt;
1 Stage IC                 1
2 Stage IIA                3
3 Stage IIB                5
4 Stage IIC               18
5 Stage IIIA               7
6 Stage IIIB              17
7 Stage IIIC             311
8 Stage IV                64
9 &lt;NA&gt;                     3</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse to one row per patient</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>patient_level <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>    sample,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">figo_stage =</span> figo_stage.diagnoses,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">age =</span> age_at_earliest_diagnosis_in_years.diagnoses.xena_derived</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(age), <span class="sc">!</span><span class="fu">is.na</span>(figo_stage))</span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a>patient_level <span class="sc">|&gt;</span> </span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> figo_stage, <span class="at">y =</span> age)) <span class="sc">+</span></span>
<span id="cb77-13"><a href="#cb77-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">"skyblue"</span>) <span class="sc">+</span></span>
<span id="cb77-14"><a href="#cb77-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.4</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb77-15"><a href="#cb77-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb77-16"><a href="#cb77-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Age Distribution by Tumor Stage (Patient Level)"</span>,</span>
<span id="cb77-17"><a href="#cb77-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Tumor Stage"</span>,</span>
<span id="cb77-18"><a href="#cb77-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Age at Diagnosis"</span></span>
<span id="cb77-19"><a href="#cb77-19" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/04_age_tumorstage.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<p>The graph shows how age at diagnosis varies across different tumor stages. Stage IIB, IIC, IIIC, and IV show wide spreads in age, meaning patient ages vary a lot within those groups. As seen in the table above, there are few data points for stages IC, IIA, IIB and IIIA. There are some outliers, especially in Stage IIIB.</p>
</section>
<section id="boxplot-across-all-samples-of-a-set-of-genes" class="level3">
<h3 class="anchored" data-anchor-id="boxplot-across-all-samples-of-a-set-of-genes">Boxplot across all samples of a set of genes</h3>
<p>The graph displays boxplots representing the distribution of gene expression levels for five randomly selected genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pick 5 random genes</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>random_genes <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Ensembl_ID) <span class="sc">|&gt;</span> </span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">5</span>) <span class="sc">|&gt;</span> </span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID)</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the data for the chosen genes</span></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>subset_data <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> random_genes)</span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>subset_data <span class="sc">|&gt;</span> </span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> log2_expression)) <span class="sc">+</span></span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">"skyblue"</span>) <span class="sc">+</span></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.4</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Log2 Expression of Randomly Selected Genes"</span>,</span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Gene (Ensembl ID)"</span>,</span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Log2 Expression"</span></span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a>  ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/04_log2_expression.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<p>The graph provides a glance on the gene expression of a selection of 5 genes.</p>
</section>
<section id="histogram-of-gene-expression-levels" class="level3">
<h3 class="anchored" data-anchor-id="histogram-of-gene-expression-levels">Histogram of gene expression levels</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span> </span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2_expression)) <span class="sc">+</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">50</span>, <span class="at">fill =</span> <span class="st">"skyblue"</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">mean</span>(log2_expression)), <span class="at">color =</span> <span class="st">"red"</span>,</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gene Expression Distribution"</span>,</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Expression Level"</span>,</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Count"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-70-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/04_histogram_expression.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<p>The mean log2 gene expression level is at around 1.8. It seems to be normally distributed.</p>
</section>
</section>
<section id="loading-libraries-and-data-1" class="level2">
<h2 class="anchored" data-anchor-id="loading-libraries-and-data-1">Loading libraries and data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load 3 tables</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>phenotype <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/phenotype_clean.rds"</span>)</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/gen_exp_clean.rds"</span>)</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analysis-of-the-distribution-of-variables" class="level2">
<h2 class="anchored" data-anchor-id="analysis-of-the-distribution-of-variables">Analysis of the distribution of variables</h2>
<p>In this section we will try to find phenotype variables that can be relevant for further analysis. We grouped the joined data set by sample and plotted phenotype variables to see if we could observe any underlying correlation.</p>
<p><strong>FIGO Stage Distribution</strong></p>
<p>The bar plot shows the distribution of patients across FIGO stages. From the plot, it is clear that the majority of patients are classified as stage IIIC, while the numbers in other stages (stage IC, IIA, IIB, IIC, IIIA, IIIB, IIIC and IV) are considerably lower. This uneven distribution suggests that most patients in this dataset were diagnosed at an advanced stage of disease, which is common for ovarian cancers due to late detection. Because the dataset is heavily skewed toward Stage III, it is difficult to draw meaningful comparisons between stages or to identify trends related to clinical outcomes across the full spectrum of FIGO stages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">|&gt;</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">|&gt;</span> <span class="co"># because we count each sample once</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(figo_stage.diagnoses)) <span class="sc">|&gt;</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> figo_stage.diagnoses)) <span class="sc">+</span></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of FIGO stages"</span>,</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"FIGO stage"</span>,</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of patients"</span>)</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Save to file</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/figostages_analysis1.png"</span>, plot1, <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">4</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Alive or death status</strong></p>
<p>The bar plot shows the distribution of patients classified as “alive” or “dead” in the dataset. While the plot provides a visual overview of survival status, the interpretation is limited because we lack detailed information about the causes of death, the timing of events, or the follow-up duration. Without this contextual information, it is not possible to draw meaningful conclusions regarding survival patterns, treatment effectiveness, or disease progression. The data simply indicate whether a patient was recorded as alive or dead at a certain point, but do not capture the underlying clinical or temporal context necessary for a robust survival analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">|&gt;</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">|&gt;</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> alive)) <span class="sc">+</span></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of alive patients"</span>,</span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"alive"</span>,</span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of patients"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-87-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="biological-relevance-of-most-variant-genes" class="level2">
<h2 class="anchored" data-anchor-id="biological-relevance-of-most-variant-genes">Biological relevance of most variant genes</h2>
<p>In this section we will extract the top 10 genes by variance of FPKM, found their biological signification and plotted expression values across all samples.</p>
<section id="finding-most-variant-genes" class="level3">
<h3 class="anchored" data-anchor-id="finding-most-variant-genes">Finding most variant genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute variance of expression per gene</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>gene_var <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">|&gt;</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">var_expression =</span> <span class="fu">var</span>(<span class="fu">c_across</span>(<span class="sc">-</span>Ensembl_ID), <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.before =</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(var_expression))</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract top 10 genes</span></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> gene_var <span class="sc">|&gt;</span> </span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">|&gt;</span> </span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID) <span class="co"># Makes genes_of_interest a character vector with genes names</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>genes_of_interest</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ENSG00000085465.13" "ENSG00000099953.10" "ENSG00000105664.11"
 [4] "ENSG00000102109.9"  "ENSG00000075891.22" "ENSG00000103355.13"
 [7] "ENSG00000022556.16" "ENSG00000083782.8"  "ENSG00000002726.21"
[10] "ENSG00000101144.13"</code></pre>
</div>
</div>
<p>The genes are listed using their Ensembl ID, which is a unique identifier assigned by the Ensembl projects. we will convert them to gene symbols to make them more understandable.</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Ensembl_ID</th>
<th>Gene symbol</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000211592.8</td>
<td>IGKC</td>
</tr>
<tr class="even">
<td>ENSG00000211896.7</td>
<td>IGHG1</td>
</tr>
<tr class="odd">
<td>ENSG00000239951.1</td>
<td>IGKV3-20</td>
</tr>
<tr class="even">
<td>ENSG00000211598.2</td>
<td>IGKV4-1</td>
</tr>
<tr class="odd">
<td>ENSG00000253755.1</td>
<td>IGHGP</td>
</tr>
<tr class="even">
<td>ENSG00000211673.2</td>
<td>IGLV3-1</td>
</tr>
<tr class="odd">
<td>ENSG00000243466.1</td>
<td>IGKV1-5</td>
</tr>
<tr class="even">
<td>ENSG00000157005.4</td>
<td>SST</td>
</tr>
<tr class="odd">
<td>ENSG00000211677.2</td>
<td>IGLC2</td>
</tr>
<tr class="even">
<td>ENSG00000241351.3</td>
<td>IGKV3-11</td>
</tr>
</tbody>
</table>
<p>These are all immunoglobulin genes, except for SST gene. They are expressed by B cells and plasma cells, not by tumor cells. Their high variability likely reflects differences in immune infiltration across ovarian tumor samples.</p>
<p>This suggests that immune involvement varies substantially among patients, which is consistent with known heterogeneity in the tumor microenvironment. In other words, different patients’ tumors contain different levels of immune cells, a result that is expected in ovarian cancer.</p>
</section>
<section id="expression-across-samples" class="level3">
<h3 class="anchored" data-anchor-id="expression-across-samples">Expression across samples</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot of expression</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>gen_exp <span class="sc">|&gt;</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> genes_of_interest) <span class="sc">|&gt;</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Ensembl_ID =</span> <span class="fu">factor</span>(Ensembl_ID, <span class="at">levels =</span> genes_of_interest)) <span class="sc">|&gt;</span></span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Ensembl_ID,</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">|&gt;</span></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Ensembl_ID, <span class="at">y =</span> expression)) <span class="sc">+</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(</span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">45</span>,        </span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="dv">1</span>,         </span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">6</span>           </span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb93-18"><a href="#cb93-18" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb93-19"><a href="#cb93-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb93-20"><a href="#cb93-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Expression of the 10 most variable genes across patients"</span>,</span>
<span id="cb93-21"><a href="#cb93-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Expression"</span></span>
<span id="cb93-22"><a href="#cb93-22" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-90-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="relationship-with-phenotype-variables" class="level3">
<h3 class="anchored" data-anchor-id="relationship-with-phenotype-variables">Relationship with phenotype variables</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Joining the genes of interest expressions with phenotype</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>joined <span class="ot">&lt;-</span> gen_exp <span class="sc">|&gt;</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> genes_of_interest) <span class="sc">|&gt;</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Ensembl_ID,</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expr"</span>) <span class="sc">|&gt;</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">|&gt;</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">immune_score =</span> <span class="fu">mean</span>(expr)) <span class="sc">|&gt;</span></span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(phenotype, <span class="at">by =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After several attempts, we found no clear relationship between these genes expressions and any patients characteristics, such as age or alive status. These plots show the lack of association between the variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune genes vs age</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(joined, <span class="fu">aes</span>(<span class="at">x =</span> immune_score, <span class="at">y =</span> age_at_index.demographic)) <span class="sc">+</span> </span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune genese vs alive/dead</span></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(joined, <span class="fu">aes</span>(<span class="at">x =</span> immune_score, <span class="at">y =</span> alive)) <span class="sc">+</span> </span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune genes vs age</span></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> joined <span class="sc">|&gt;</span> </span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">race.demographic =</span> <span class="fu">fct_relevel</span>(race.demographic, <span class="st">"not reported"</span>, <span class="at">after =</span> <span class="cn">Inf</span>), <span class="co"># NAs at bottom</span></span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">race.demographic =</span> <span class="fu">case_when</span>( <span class="co"># split long labels over 2 lines</span></span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>      race.demographic <span class="sc">==</span> <span class="st">"native hawaiian or other pacific islander"</span> <span class="sc">~</span> </span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"native hawaiian</span><span class="sc">\n</span><span class="st">or other pacific islander"</span>,</span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>      race.demographic <span class="sc">==</span> <span class="st">"american indian or alaska native"</span> <span class="sc">~</span></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">"american indian</span><span class="sc">\n</span><span class="st">or alaska native"</span>,</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>      race.demographic <span class="sc">==</span> <span class="st">"black or african american"</span> <span class="sc">~</span></span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>        <span class="st">"black </span><span class="sc">\n</span><span class="st">or african american"</span>,</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> race.demographic)) <span class="sc">|&gt;</span> </span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> immune_score, <span class="at">y =</span> race.demographic)) <span class="sc">+</span> </span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Immune score by ethnicity"</span>,</span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Immune score"</span>,</span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Ethnicity"</span></span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)</span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>plot1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-92-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-92-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>plot3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-92-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save to file</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/IS_ethnicity_analysis1.png"</span>, plot3, <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">4</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="pca-analysis" class="level1">
<h1>PCA analysis</h1>
<section id="aim" class="level2">
<h2 class="anchored" data-anchor-id="aim">Aim</h2>
<p>We explore the gene expression dataset using Principal Component Analysis (PCA).</p>
<p>We work with the augmented tidy dataset created in 03_augment.qmd and check whether patients separate in the PCA space depending on clinical variables such as the alive status and age category.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-augmented-data" class="level2">
<h2 class="anchored" data-anchor-id="load-augmented-data">Load augmented data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/data_augmented.rds"</span>)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>data_augmented <span class="sc">|&gt;</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 53
   sample     Ensembl_ID expression race.demographic ethnicity.demographic alive
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;                 &lt;lgl&gt;
 1 TCGA-61-2… ENSG00000…       3.73 white            not reported          TRUE 
 2 TCGA-09-0… ENSG00000…       1.81 white            not hispanic or lati… FALSE
 3 TCGA-29-1… ENSG00000…       1.98 white            not reported          FALSE
 4 TCGA-13-1… ENSG00000…       6.07 white            not hispanic or lati… FALSE
 5 TCGA-25-1… ENSG00000…       2.36 white            not hispanic or lati… FALSE
 6 TCGA-04-1… ENSG00000…       6.06 white            not hispanic or lati… TRUE 
 7 TCGA-25-1… ENSG00000…       3.28 white            not hispanic or lati… FALSE
 8 TCGA-09-2… ENSG00000…       2.68 white            not hispanic or lati… TRUE 
 9 TCGA-13-0… ENSG00000…       4.91 white            not hispanic or lati… TRUE 
10 TCGA-13-1… ENSG00000…       2.31 white            not hispanic or lati… FALSE
# ℹ 47 more variables: age_at_index.demographic &lt;dbl&gt;,
#   days_to_birth.demographic &lt;dbl&gt;, year_of_birth.demographic &lt;dbl&gt;,
#   days_to_death.demographic &lt;dbl&gt;, year_of_death.demographic &lt;dbl&gt;,
#   code.tissue_source_site &lt;chr&gt;, name.tissue_source_site &lt;chr&gt;,
#   bcr_id.tissue_source_site &lt;chr&gt;, notes.annotations &lt;chr&gt;,
#   submitter_id.annotations &lt;chr&gt;, classification.annotations &lt;chr&gt;,
#   created_datetime.annotations &lt;chr&gt;, updated_datetime.annotations &lt;chr&gt;, …</code></pre>
</div>
</div>
</section>
<section id="select-most-variable-genes" class="level2">
<h2 class="anchored" data-anchor-id="select-most-variable-genes">Select most variable genes</h2>
<p>Since PCA works best with genes that vary a lot across samples, we select the top 500 most variable genes based on the variance of log2_expression.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>gene_var <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Ensembl_ID) <span class="sc">|&gt;</span></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">var_log2 =</span> <span class="fu">var</span>(log2_expression, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups  =</span> <span class="st">"drop"</span>) <span class="sc">|&gt;</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(var_log2))</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>gene_var <span class="sc">|&gt;</span></span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 2
   Ensembl_ID         var_log2
   &lt;chr&gt;                 &lt;dbl&gt;
 1 ENSG00000022556.16    0.978
 2 ENSG00000075891.22    0.900
 3 ENSG00000083782.8     0.861
 4 ENSG00000103355.13    0.834
 5 ENSG00000002726.21    0.753
 6 ENSG00000019102.12    0.723
 7 ENSG00000043355.12    0.723
 8 ENSG00000060718.22    0.712
 9 ENSG00000105664.11    0.659
10 ENSG00000085465.13    0.654</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> gene_var <span class="sc">|&gt;</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">500</span>) <span class="sc">|&gt;</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Ensembl_ID)</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(top_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 500</code></pre>
</div>
</div>
</section>
<section id="build-expression-matrix-for-pca" class="level2">
<h2 class="anchored" data-anchor-id="build-expression-matrix-for-pca">Build expression matrix for PCA</h2>
<p>Here, we create a wide matrix with one row per sample (patient) and one column per gene (for the selected 500 genes). We also extract the clinical variables so we can link them to the PCA results later.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>expr_wide <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> top_genes) <span class="sc">|&gt;</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, Ensembl_ID, log2_expression) <span class="sc">|&gt;</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from  =</span> Ensembl_ID,<span class="at">values_from =</span> log2_expression)</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>pheno_pca <span class="ot">&lt;-</span> data_augmented <span class="sc">|&gt;</span></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, alive, age_cat, age_at_index.demographic) <span class="sc">|&gt;</span></span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>expr_wide <span class="ot">&lt;-</span> expr_wide <span class="sc">|&gt;</span></span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pheno_pca, <span class="at">by =</span> <span class="st">"sample"</span>)</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>expr_wide <span class="sc">|&gt;</span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 504
  sample           ENSG00000101489.20 ENSG00000108176.15 ENSG00000108602.18
  &lt;chr&gt;                         &lt;dbl&gt;              &lt;dbl&gt;              &lt;dbl&gt;
1 TCGA-24-1104-01A              0.940              0.457              0.606
2 TCGA-25-1320-01A              1.20               1.24               0.711
3 TCGA-24-1850-01A              0.606              0.808              0.279
4 TCGA-61-2101-01A              1.39               0.933              1.01 
5 TCGA-23-1111-01A              0.729              0.612              1.28 
# ℹ 500 more variables: ENSG00000101298.15 &lt;dbl&gt;, ENSG00000043355.12 &lt;dbl&gt;,
#   ENSG00000010932.17 &lt;dbl&gt;, ENSG00000010671.16 &lt;dbl&gt;,
#   ENSG00000069812.11 &lt;dbl&gt;, ENSG00000029559.7 &lt;dbl&gt;,
#   ENSG00000005844.19 &lt;dbl&gt;, ENSG00000100156.10 &lt;dbl&gt;,
#   ENSG00000090932.11 &lt;dbl&gt;, ENSG00000105997.22 &lt;dbl&gt;,
#   ENSG00000100583.4 &lt;dbl&gt;, ENSG00000088992.18 &lt;dbl&gt;,
#   ENSG00000096006.12 &lt;dbl&gt;, ENSG00000008311.15 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="run-pca" class="level2">
<h2 class="anchored" data-anchor-id="run-pca">Run PCA</h2>
<p>We extract the numeric expression data, centre and scale it, and run PCA using prcomp().</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> expr_wide <span class="sc">|&gt;</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>sample, <span class="sc">-</span>alive, <span class="sc">-</span>age_cat, <span class="sc">-</span>age_at_index.demographic) <span class="sc">|&gt;</span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr_mat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 429 500</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>pca_res <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x      =</span> expr_mat,</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">center =</span> <span class="cn">TRUE</span>,</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>var_explained <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">component =</span> <span class="fu">seq_along</span>(pca_res<span class="sc">$</span>sdev),</span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">var_explained =</span> pca_res<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pca_res<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a>var_explained <span class="sc">|&gt;</span>  </span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 2
  component var_explained
      &lt;int&gt;         &lt;dbl&gt;
1         1        0.151 
2         2        0.0923
3         3        0.0462
4         4        0.0327
5         5        0.0291</code></pre>
</div>
</div>
</section>
<section id="scree-plot" class="level2">
<h2 class="anchored" data-anchor-id="scree-plot">Scree plot</h2>
<p>This plot shows how much variance is explained by each principal component.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>scree_df <span class="ot">&lt;-</span> var_explained <span class="sc">|&gt;</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transmute</span>(</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>    component,</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">PC =</span> stringr<span class="sc">::</span><span class="fu">str_c</span>(<span class="st">"PC"</span>, component),</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">variance =</span> var_explained) <span class="sc">|&gt;</span></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">PC =</span> forcats<span class="sc">::</span><span class="fu">fct_reorder</span>(PC, component))</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>scree_plot <span class="ot">&lt;-</span> scree_df <span class="sc">|&gt;</span></span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">15</span>) <span class="sc">|&gt;</span></span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y =</span> variance, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Variance explained by principal components"</span>,</span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Principal component"</span>,</span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Proportion of variance explained"</span>)</span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>scree_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-108-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"../results/06_pca_scree.png"</span>,</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> scree_plot,</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The scree plot shows that most of the global variation comes from PC1. If there is a biological separation between patient groups, it should appear mainly along PC1 or PC2.</p>
</section>
<section id="pc1-vs-pc2-coloured-by-clinical-variables" class="level2">
<h2 class="anchored" data-anchor-id="pc1-vs-pc2-coloured-by-clinical-variables">PC1 vs PC2 coloured by clinical variables</h2>
<p>Here, we plot PC1 vs PC2 and colour the samples by alive status and shape by age category.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(pca_res<span class="sc">$</span>x, <span class="at">rownames =</span> <span class="st">"row_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(expr_wide <span class="sc">|&gt;</span> <span class="fu">select</span>(sample, alive, age_cat, age_at_index.demographic)) <span class="sc">|&gt;</span></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(sample)</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">|&gt;</span></span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 434
  sample      row_id   PC1   PC2    PC3    PC4    PC5    PC6    PC7    PC8   PC9
  &lt;chr&gt;       &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 TCGA-24-11… 1      -5.05  8.85 -2.47  -5.44   0.474 -2.79   2.90  -0.962  2.08
2 TCGA-25-13… 2      -8.54 -9.19 -0.236 -0.604 -0.518  0.411 -3.01  -0.150  4.25
3 TCGA-24-18… 3      -5.83 -6.17 -0.951  7.97   2.08   1.77  -0.791 -1.24   2.22
4 TCGA-61-21… 4       1.47 -5.07 -8.05  -7.54   0.505  1.51  -3.28  -2.69  -1.15
5 TCGA-23-11… 5       6.05 -3.13  0.131 -5.00   3.24  -0.788  0.841 -2.12   2.10
# ℹ 423 more variables: PC10 &lt;dbl&gt;, PC11 &lt;dbl&gt;, PC12 &lt;dbl&gt;, PC13 &lt;dbl&gt;,
#   PC14 &lt;dbl&gt;, PC15 &lt;dbl&gt;, PC16 &lt;dbl&gt;, PC17 &lt;dbl&gt;, PC18 &lt;dbl&gt;, PC19 &lt;dbl&gt;,
#   PC20 &lt;dbl&gt;, PC21 &lt;dbl&gt;, PC22 &lt;dbl&gt;, PC23 &lt;dbl&gt;, PC24 &lt;dbl&gt;, PC25 &lt;dbl&gt;,
#   PC26 &lt;dbl&gt;, PC27 &lt;dbl&gt;, PC28 &lt;dbl&gt;, PC29 &lt;dbl&gt;, PC30 &lt;dbl&gt;, PC31 &lt;dbl&gt;,
#   PC32 &lt;dbl&gt;, PC33 &lt;dbl&gt;, PC34 &lt;dbl&gt;, PC35 &lt;dbl&gt;, PC36 &lt;dbl&gt;, PC37 &lt;dbl&gt;,
#   PC38 &lt;dbl&gt;, PC39 &lt;dbl&gt;, PC40 &lt;dbl&gt;, PC41 &lt;dbl&gt;, PC42 &lt;dbl&gt;, PC43 &lt;dbl&gt;,
#   PC44 &lt;dbl&gt;, PC45 &lt;dbl&gt;, PC46 &lt;dbl&gt;, PC47 &lt;dbl&gt;, PC48 &lt;dbl&gt;, PC49 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>pc1_label <span class="ot">&lt;-</span> var_explained <span class="sc">|&gt;</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">|&gt;</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> stringr<span class="sc">::</span><span class="fu">str_c</span>(<span class="st">"PC1 ("</span>, scales<span class="sc">::</span><span class="fu">percent</span>(var_explained), <span class="st">")"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(label)</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>pc2_label <span class="ot">&lt;-</span> var_explained <span class="sc">|&gt;</span></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> stringr<span class="sc">::</span><span class="fu">str_c</span>(<span class="st">"PC2 ("</span>, scales<span class="sc">::</span><span class="fu">percent</span>(var_explained), <span class="st">")"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(label)</span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>pc1_pc2_plot <span class="ot">&lt;-</span> scores <span class="sc">|&gt;</span></span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1,</span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> PC2,</span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> alive,</span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> age_cat)) <span class="sc">+</span></span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA on top 500 most variable genes"</span>,</span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Colour: alive status, shape: age category"</span>,</span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> pc1_label,</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> pc2_label)</span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a>pc1_pc2_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-111-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"../results/06_pca_pc1_pc2.png"</span>,</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> pc1_pc2_plot,</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>There is no clear separation between the groups in PC1–PC2 space. Both the alive/deceased groups and the age categories are mixed together, so these variables do not seem to drive the main variation.</p>
</section>
<section id="conclusion" class="level2">
<h2 class="anchored" data-anchor-id="conclusion">Conclusion</h2>
<p>Running PCA on the 500 most variable genes shows that the major variation in gene expression is not linked to the clinical variables considered here. PC1 and PC2 capture global transcriptomic patterns, but these patterns do not reflect differences in survival status or age.</p>
</section>
</section>
<section id="section" class="level1">
<h1></h1>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>