---
title: "04 - describe"
format: html
editor: visual
execute:
  cache: false
---

### 0.Loading libraries and data

```{r}
library("tidyverse")
```

```{r}
# Load 3 tables
phenotype_clean <- readRDS("../data/phenotype_clean.rds")
gen_exp_clean <- readRDS("../data/gen_exp_clean.rds")
data_augmented <- readRDS("../data/data_augmented.rds")
```

## 1. General information

The analysis will be based on the article *Comparison of gene expression patterns across twelve tumor types identifies a cancer supercluster characterized by TP53 mutations and cell cycle defects*. The datasets used in the article are from the The Cancer Genome Atlas (TCGA) ovarian cancer collection. In the article, compares gene-expression profiles across multiple tumor types to identify transcriptional superclusters.

Two different data sets are used:

-   A clinical dataset: TCGA-OV.clinical.tsv

-   A gene expression dataset: TCGA-OV.star_fpkm.tsv

The clinical and transcriptomic datasets for the TCGA ovarian cancer cohort will be used to search for gene expression associations with clinical variables and outcomes.

## 2. Dataset descriptions

### Clinical dataset

-   Format: Tab-separated values (TSV) file.

-   Contains clinical variables from the TCGA ovarian cancer cohort. As it contains observations about the patients, it is called `phenotype` in this project.

-   Size: There are 632 patients as rows and 84 clinical variables as columns.

-   Key variables and data types: The following variables are most relevant for the analysis, - `alive` (bool),

    -   `figo_stage.diagnoses` (chr),
    -   `morphology.diagnoses` (chr, the first part encodes the tumor type, the second part encodes the malignancy),
    -   `age_at_earliest_diagnosis_in_years.diagnoses.xena_derived` (numeric),
    -   `treatment_type.treatments.diagnoses` (chr, indicating the type of treatement),

-   Needed cleaning: The dataset contains a lot of redundant variables (e.g. many instances of a variable indicating the ID), many have been combined into a single one or have been removed altogether. Also the variable containing information about the gender has been removed as all patients with ovarian cancer are female. Some variables contain annotations to the patients, but as most are empty and irrelevant for the analysis they have been removed as well.

-   Size after cleaning: 624 patients on the rows, 52 clinical variables on the columns.

Underneath, 10 randomly selected rows of the set are shown to understand the structure:

```{r}

phenotype_clean |> slice_sample(n=10)

```

### Gene expression dataset

-   Format: Tab-separated values (TSV) file.

-   Contains gene expression data (FPKM) from the TCGA ovarian cancer cohort.

-   The values are FPKM (Fragments Per Kilobase of transcript per Million mapped reads) which is a normalized gene-expression metric.

-   Size: There are 60660 genes on the rows and 430 samples on the columns.

-   Data types: numeric nonâ€negative values

-   Needed cleaning: Some genes have a null expression level for all samples, these genes have been removed from the dataset. To further reduce the number of genes and keep only the most relevant ones, the summed expression level over all samples for each gene has been computed and the relevant genes were kept based on a threshold. A variable representing this sum of the expression levels for each gene has been added.

-   Size after cleaning: 14157 genes on the rows, 431 variables on the columns.

Underneath, 10 randomly selected rows of the set are shown to understand the structure:

```{r}

gen_exp_clean |> slice_sample(n = 10)

```

### Joining of the datasets

These two datasets have been combined to one by matching the patient IDs (samples). As the two datasets do not show the same amount of samples (the phenotype dataset is bigger than the gene expression dataset), the IDs that could not be resolved between the two sets have been removed. This leaves only patients who have recorded gene expression data (no gene expression data is lost).

Two variables have been added: - `log2_expression` (numeric, gene expression on logarithmic scale) - `age_cat` (chr, age category the patient belongs to: young, middle, senior)

In addition to the key variables mentioned for the phenotype (clinical) data and the gene expression values from the gene expression data, these two variables will allow for easier visualization and analysis.

In the `figo_stage.diagnoses` variable, the tumor stages IA, IB and the majority of IC have disappeared after joining. This is due to the fact that it it rare to have gene expression data for early stage cancers.

Underneath, the 10 first rows of the joined set are shown to understand the structure:

```{r}

data_augmented |> head(10)

```

## 3. Quality and intergrity

In the phenotype datasets, a few columns containing only NAs have been removed. Some of the variables contained a label "Not Reported" instead of NA. As described above, many variables described redundant information and needed to be removed.

The gene expression dataset is of good quality and integrity, the only correction needed was to remove uninformative genes.

Overall the data seems of good quality without aberrant values or errors in the writing of the sets.

## 4. Visualisation

The following graphs will allow to visualize the content of the data set.

A list of the variables contained in the full dataset:

```{r}

data_augmented |> glimpse()

```

```{r}

data_augmented_numeric_summary <- data_augmented |> 
  select(where(is.numeric)) |> 
  summarise(across(everything(),
                   list(mean = ~mean(.x, na.rm = TRUE),
                        median = ~median(.x, na.rm = TRUE),
                        sd = ~sd(.x, na.rm = TRUE),
                        min = ~min(.x, na.rm = TRUE),
                        max = ~max(.x, na.rm = TRUE),
                        missing = ~sum(is.na(.x))),
                   .names = "{.col}_{.fn}")) 

data_augmented_numeric_summary

```

### Distribution of the age at diagnosis

The graph is a histogram showing the distribution of patient age at diagnosis.

```{r}

patient_level <- data_augmented |> 
  distinct(
    sample,
    age = age_at_earliest_diagnosis_in_years.diagnoses.xena_derived
  ) |> 
  filter(!is.na(age))

patient_level |> 
  ggplot(aes(x = age)) +
  geom_histogram(bins = 30, fill = "skyblue", color = "white") +
  geom_vline(aes(xintercept = mean(age)), color = "red",
             linetype = "dashed", size = 1) +
  theme_minimal() +
  labs(
    title = "Distribution of Age at Diagnosis (Patient Level)",
    x = "Age",
    y = "Number of Patients"
  )
ggsave("../doc/images/age.png")
```

The age of patients at diagnosis seems to be normally distributed with a mean (red line) at around \~60 years.

### Age distribution at tumor stage

Tumor stages IA, IB and the majority of IC have disappeared after joining the initial two datasets. Most patients, out of those which also have gene expression data, have ovarian cancer stage IIIC. There are very few data points for stages IC, IIA, IIB and IIIA.

```{r}

data_augmented |> 
  distinct(sample, figo_stage.diagnoses) |> 
  count(figo_stage.diagnoses)

```

```{r}

# Collapse to one row per patient
patient_level <- data_augmented |> 
  distinct(
    sample,
    figo_stage = figo_stage.diagnoses,
    age = age_at_earliest_diagnosis_in_years.diagnoses.xena_derived
  ) |> 
  filter(!is.na(age), !is.na(figo_stage))


patient_level |> 
  ggplot(aes(x = figo_stage, y = age)) +
  geom_boxplot(fill = "skyblue") +
  geom_jitter(color="black", size=0.4, alpha=0.5) +
  theme_minimal() +
  labs(
    title = "Age Distribution by Tumor Stage (Patient Level)",
    x = "Tumor Stage",
    y = "Age at Diagnosis"
  )


```

The graph shows how age at diagnosis varies across different tumor stages. Stage IIB, IIC, IIIC, and IV show wide spreads in age, meaning patient ages vary a lot within those groups. As seen in the table above, there are few data points for stages IC, IIA, IIB and IIIA. There are some outliers, especially in Stage IIIB.

### Boxplot across all samples of a set of genes

The graph displays boxplots representing the distribution of gene expression levels for five randomly selected genes.

```{r}

# Pick 5 random genes
random_genes <- data_augmented |> 
  distinct(Ensembl_ID) |> 
  sample_n(5) |> 
  pull(Ensembl_ID)

# Filter the data for the chosen genes
subset_data <- data_augmented |> 
  filter(Ensembl_ID %in% random_genes)

subset_data |> 
  ggplot(aes(x = Ensembl_ID, y = log2_expression)) +
  geom_boxplot(fill = "skyblue") +
  geom_jitter(color="black", size=0.4, alpha=0.5) +
  theme_minimal() +
  labs(
    title = "Log2 Expression of Randomly Selected Genes",
    x = "Gene (Ensembl ID)",
    y = "Log2 Expression"
  ) 




```

The graph provides a glance on the gene expression of a selection of 5 genes.

### Histogram of gene expression levels

```{r}

data_augmented |> 
  ggplot(aes(x = log2_expression)) +
  geom_histogram(bins = 50, fill = "skyblue", color = "white") +
  geom_vline(aes(xintercept = mean(log2_expression)), color = "red",
             linetype = "dashed", size = 1) +
  theme_minimal() +
  labs(title = "Gene Expression Distribution",
       x = "Expression Level",
       y = "Count")

```

The mean log2 gene expression level is at around 1.8. It seems to be normally distributed.
